{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import sys\n",
    "import pandas \n",
    "\n",
    "# import utils.rawgit()\n",
    "rawgit = 'https://raw.githubusercontent.com/' \n",
    "# local imports\n",
    "sys.path.insert(0, './data/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Read DO Slim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>doid_id</th>\n",
       "      <th>doid_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DOID:2531</td>\n",
       "      <td>Hematologic cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DOID:1319</td>\n",
       "      <td>Brain cancer</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     doid_id           doid_name\n",
       "0  DOID:2531  Hematologic cancer\n",
       "1  DOID:1319        Brain cancer"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "commit = '72614ade9f1cc5a5317b8f6836e1e464b31d5587'\n",
    "#url = utils.rawgit('dhimmel', 'disease-ontology', commit, 'data/slim-terms.tsv')\n",
    "url = rawgit + 'dhimmel/' + 'disease-ontology/' + commit + '/data/slim-terms.tsv'\n",
    "disease_df = pandas.read_table(url)\n",
    "disease_df = disease_df.rename(columns={'doid': 'doid_id', 'name': 'doid_name'})\n",
    "disease_df = disease_df[['doid_id', 'doid_name']]\n",
    "disease_df.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Read Entrez Gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>entrez_gene_id</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   entrez_gene_id gene_symbol\n",
       "0               1        A1BG\n",
       "1               2         A2M"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#commit = '6e133f9ef8ce51a4c5387e58a6cc97564a66cec8'\n",
    "##url = utils.rawgit('dhimmel', 'entrez-gene', commit, 'data/genes-human.tsv')\n",
    "#url = rawgit + 'dhimmel/' + 'entrez-gene/' + commit + '/data/genes-human.tsv'\n",
    "gene_df = pandas.read_table('../entrez-gene/data/genes-human.tsv')\n",
    "gene_df = gene_df[gene_df.type_of_gene == 'protein-coding']\n",
    "gene_df = gene_df.rename(columns={'GeneID': 'entrez_gene_id', 'Symbol': 'gene_symbol'})\n",
    "gene_df = gene_df[['entrez_gene_id', 'gene_symbol']]\n",
    "gene_df.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Read Datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orphanet_code</th>\n",
       "      <th>orphanet_term</th>\n",
       "      <th>orphanet_gene_id</th>\n",
       "      <th>gene_orphanumber</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>dga_type</th>\n",
       "      <th>dga_status</th>\n",
       "      <th>entrez_gene_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DOID:0050461</td>\n",
       "      <td>Aspartylglucosaminuria</td>\n",
       "      <td>15470</td>\n",
       "      <td>119513</td>\n",
       "      <td>aspartylglucosaminidase</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>AGA</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>175.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DOID:0050441</td>\n",
       "      <td>Multiple sulfatase deficiency</td>\n",
       "      <td>15570</td>\n",
       "      <td>119899</td>\n",
       "      <td>sulfatase modifying factor 1</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>SUMF1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>285362.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  orphanet_code                  orphanet_term  orphanet_gene_id  \\\n",
       "0  DOID:0050461         Aspartylglucosaminuria             15470   \n",
       "1  DOID:0050441  Multiple sulfatase deficiency             15570   \n",
       "\n",
       "   gene_orphanumber                     gene_name                  gene_type  \\\n",
       "0            119513       aspartylglucosaminidase  gene with protein product   \n",
       "1            119899  sulfatase modifying factor 1  gene with protein product   \n",
       "\n",
       "  gene_symbol                                 dga_type dga_status  \\\n",
       "0         AGA  Disease-causing germline mutation(s) in   Assessed   \n",
       "1       SUMF1  Disease-causing germline mutation(s) in   Assessed   \n",
       "\n",
       "   entrez_gene_id  \n",
       "0           175.0  \n",
       "1        285362.0  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Orphanet\n",
    "#orphanet_df = pandas.read_table('../data/orphanet-disease-gene.tsv')\n",
    "orphanet_df = pandas.read_table('../data/orphanet-mondo-disease-gene.tsv')\n",
    "orphanet_df = orphanet_df.merge(gene_df, on='gene_symbol', how='left')\n",
    "orphanet_df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'20455 genes in entrez and 3641 genes in orphanet'"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'{} genes in entrez and {} genes in orphanet'.format(len(gene_df.gene_symbol.unique()), len(orphanet_df.gene_symbol.unique()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gene_types: ['gene with protein product' 'Non-coding RNA' 'locus']\n",
      "dga_status: ['Assessed' 'Not yet assessed']\n"
     ]
    }
   ],
   "source": [
    "print('gene_types: {}'.format(orphanet_df.gene_type.unique()))\n",
    "print('dga_status: {}'.format(orphanet_df.dga_status.unique()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orphanet_code</th>\n",
       "      <th>orphanet_term</th>\n",
       "      <th>orphanet_gene_id</th>\n",
       "      <th>gene_orphanumber</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>dga_type</th>\n",
       "      <th>dga_status</th>\n",
       "      <th>entrez_gene_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>DOID:3687</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>16471</td>\n",
       "      <td>123512</td>\n",
       "      <td>mitochondrially encoded cytochrome c oxidase I</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>DOID:3687</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>16472</td>\n",
       "      <td>123516</td>\n",
       "      <td>mitochondrially encoded cytochrome c oxidase II</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-CO2</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>DOID:3687</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>16473</td>\n",
       "      <td>123520</td>\n",
       "      <td>mitochondrially encoded cytochrome c oxidase III</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-CO3</td>\n",
       "      <td>Candidate gene tested in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>DOID:3687</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>16478</td>\n",
       "      <td>123537</td>\n",
       "      <td>mitochondrially encoded NADH:ubiquinone oxidor...</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-ND1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>DOID:3687</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>16481</td>\n",
       "      <td>123552</td>\n",
       "      <td>mitochondrially encoded NADH:ubiquinone oxidor...</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-ND4</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>DOID:3687</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>16483</td>\n",
       "      <td>123562</td>\n",
       "      <td>mitochondrially encoded NADH:ubiquinone oxidor...</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-ND5</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>DOID:3687</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>16484</td>\n",
       "      <td>123567</td>\n",
       "      <td>mitochondrially encoded NADH:ubiquinone oxidor...</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-ND6</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>DOID:3687</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>16841</td>\n",
       "      <td>138895</td>\n",
       "      <td>mitochondrially encoded tRNA leucine 1 (UUA/G)</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TL1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>DOID:3687</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>16844</td>\n",
       "      <td>138906</td>\n",
       "      <td>mitochondrially encoded tRNA glutamine</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TQ</td>\n",
       "      <td>Candidate gene tested in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>DOID:3687</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>17473</td>\n",
       "      <td>160307</td>\n",
       "      <td>mitochondrially encoded tRNA tryptophan</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TW</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>DOID:3687</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>17671</td>\n",
       "      <td>166766</td>\n",
       "      <td>mitochondrially encoded tRNA histidine</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TH</td>\n",
       "      <td>Candidate gene tested in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>DOID:3687</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>17722</td>\n",
       "      <td>167901</td>\n",
       "      <td>mitochondrially encoded tRNA serine 1 (UCN)</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TS1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>DOID:3687</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>17723</td>\n",
       "      <td>167905</td>\n",
       "      <td>mitochondrially encoded tRNA serine 2 (AGU/C)</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TS2</td>\n",
       "      <td>Candidate gene tested in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>DOID:3687</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>17724</td>\n",
       "      <td>167909</td>\n",
       "      <td>mitochondrially encoded tRNA phenylalanine</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TF</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>DOID:11727</td>\n",
       "      <td>Facioscapulohumeral dystrophy</td>\n",
       "      <td>15901</td>\n",
       "      <td>121239</td>\n",
       "      <td>double homeobox 4 like 1</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>DUX4L1</td>\n",
       "      <td>Candidate gene tested in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>DOID:12934</td>\n",
       "      <td>Kearns-Sayre syndrome</td>\n",
       "      <td>16841</td>\n",
       "      <td>138895</td>\n",
       "      <td>mitochondrially encoded tRNA leucine 1 (UUA/G)</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TL1</td>\n",
       "      <td>Candidate gene tested in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>DOID:12934</td>\n",
       "      <td>Kearns-Sayre syndrome</td>\n",
       "      <td>17396</td>\n",
       "      <td>159773</td>\n",
       "      <td>mitochondrially encoded ATP synthase 8</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-ATP8</td>\n",
       "      <td>Candidate gene tested in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>DOID:310</td>\n",
       "      <td>MERRF</td>\n",
       "      <td>16483</td>\n",
       "      <td>123562</td>\n",
       "      <td>mitochondrially encoded NADH:ubiquinone oxidor...</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-ND5</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>DOID:310</td>\n",
       "      <td>MERRF</td>\n",
       "      <td>16841</td>\n",
       "      <td>138895</td>\n",
       "      <td>mitochondrially encoded tRNA leucine 1 (UUA/G)</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TL1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>DOID:310</td>\n",
       "      <td>MERRF</td>\n",
       "      <td>16842</td>\n",
       "      <td>138900</td>\n",
       "      <td>mitochondrially encoded tRNA lysine</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TK</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>DOID:310</td>\n",
       "      <td>MERRF</td>\n",
       "      <td>16843</td>\n",
       "      <td>138902</td>\n",
       "      <td>mitochondrially encoded 12S RNA</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>Candidate gene tested in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>DOID:310</td>\n",
       "      <td>MERRF</td>\n",
       "      <td>16844</td>\n",
       "      <td>138906</td>\n",
       "      <td>mitochondrially encoded tRNA glutamine</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TQ</td>\n",
       "      <td>Candidate gene tested in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>DOID:310</td>\n",
       "      <td>MERRF</td>\n",
       "      <td>17671</td>\n",
       "      <td>166766</td>\n",
       "      <td>mitochondrially encoded tRNA histidine</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TH</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>DOID:310</td>\n",
       "      <td>MERRF</td>\n",
       "      <td>17722</td>\n",
       "      <td>167901</td>\n",
       "      <td>mitochondrially encoded tRNA serine 1 (UCN)</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TS1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>DOID:310</td>\n",
       "      <td>MERRF</td>\n",
       "      <td>17723</td>\n",
       "      <td>167905</td>\n",
       "      <td>mitochondrially encoded tRNA serine 2 (AGU/C)</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TS2</td>\n",
       "      <td>Candidate gene tested in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>DOID:310</td>\n",
       "      <td>MERRF</td>\n",
       "      <td>17724</td>\n",
       "      <td>167909</td>\n",
       "      <td>mitochondrially encoded tRNA phenylalanine</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TF</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>DOID:310</td>\n",
       "      <td>MERRF</td>\n",
       "      <td>18493</td>\n",
       "      <td>205934</td>\n",
       "      <td>mitochondrially encoded tRNA proline</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TP</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>DOID:705</td>\n",
       "      <td>Leber hereditary optic neuropathy</td>\n",
       "      <td>16470</td>\n",
       "      <td>123508</td>\n",
       "      <td>mitochondrially encoded ATP synthase 6</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-ATP6</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>DOID:705</td>\n",
       "      <td>Leber hereditary optic neuropathy</td>\n",
       "      <td>16471</td>\n",
       "      <td>123512</td>\n",
       "      <td>mitochondrially encoded cytochrome c oxidase I</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>Candidate gene tested in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>DOID:705</td>\n",
       "      <td>Leber hereditary optic neuropathy</td>\n",
       "      <td>16473</td>\n",
       "      <td>123520</td>\n",
       "      <td>mitochondrially encoded cytochrome c oxidase III</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-CO3</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5054</th>\n",
       "      <td>Orphanet:225</td>\n",
       "      <td>Maternally-inherited diabetes and deafness</td>\n",
       "      <td>16981</td>\n",
       "      <td>140514</td>\n",
       "      <td>mitochondrially encoded tRNA glutamic acid</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TE</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5108</th>\n",
       "      <td>Orphanet:401901</td>\n",
       "      <td>Huntington disease-like syndrome due to C9ORF7...</td>\n",
       "      <td>20676</td>\n",
       "      <td>286546</td>\n",
       "      <td>chromosome 9 open reading frame 72</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>C9ORF72</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5178</th>\n",
       "      <td>Orphanet:391646</td>\n",
       "      <td>Feingold syndrome type 2</td>\n",
       "      <td>20656</td>\n",
       "      <td>285348</td>\n",
       "      <td>miR-17-92a-1 cluster host gene</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MIR17HG</td>\n",
       "      <td>Role in the phenotype of</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5205</th>\n",
       "      <td>Orphanet:397959</td>\n",
       "      <td>TCR-alpha-beta-positive T-cell deficiency</td>\n",
       "      <td>22849</td>\n",
       "      <td>401908</td>\n",
       "      <td>T-cell receptor alpha constant</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>TRAC</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5217</th>\n",
       "      <td>Orphanet:397750</td>\n",
       "      <td>Periodic paralysis with later-onset distal mot...</td>\n",
       "      <td>16470</td>\n",
       "      <td>123508</td>\n",
       "      <td>mitochondrially encoded ATP synthase 6</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-ATP6</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5218</th>\n",
       "      <td>Orphanet:397750</td>\n",
       "      <td>Periodic paralysis with later-onset distal mot...</td>\n",
       "      <td>17396</td>\n",
       "      <td>159773</td>\n",
       "      <td>mitochondrially encoded ATP synthase 8</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-ATP8</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5426</th>\n",
       "      <td>Orphanet:422526</td>\n",
       "      <td>Hereditary clear cell renal cell carcinoma</td>\n",
       "      <td>21868</td>\n",
       "      <td>326692</td>\n",
       "      <td>disrupted in renal carcinoma 3</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>DIRC3</td>\n",
       "      <td>Part of a fusion gene in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5456</th>\n",
       "      <td>Orphanet:88</td>\n",
       "      <td>Idiopathic aplastic anemia</td>\n",
       "      <td>16790</td>\n",
       "      <td>138457</td>\n",
       "      <td>telomerase RNA component</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>TERC</td>\n",
       "      <td>Major susceptibility factor in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5570</th>\n",
       "      <td>Orphanet:65684</td>\n",
       "      <td>Monomelic amyotrophy</td>\n",
       "      <td>21089</td>\n",
       "      <td>299619</td>\n",
       "      <td>chromosome 5 open reading frame 42</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>C5ORF42</td>\n",
       "      <td>Major susceptibility factor in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5647</th>\n",
       "      <td>Orphanet:75327</td>\n",
       "      <td>North Carolina macular dystrophy</td>\n",
       "      <td>23671</td>\n",
       "      <td>455932</td>\n",
       "      <td>DNase1 hypersensitivity, chromosome 6, site 1</td>\n",
       "      <td>locus</td>\n",
       "      <td>DHS6S1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5718</th>\n",
       "      <td>Orphanet:33110</td>\n",
       "      <td>Autosomal agammaglobulinemia</td>\n",
       "      <td>16247</td>\n",
       "      <td>122591</td>\n",
       "      <td>immunoglobulin heavy constant mu</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>IGHM</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5946</th>\n",
       "      <td>Orphanet:90641</td>\n",
       "      <td>Mitochondrial non-syndromic sensorineural deaf...</td>\n",
       "      <td>16471</td>\n",
       "      <td>123512</td>\n",
       "      <td>mitochondrially encoded cytochrome c oxidase I</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5947</th>\n",
       "      <td>Orphanet:90641</td>\n",
       "      <td>Mitochondrial non-syndromic sensorineural deaf...</td>\n",
       "      <td>16843</td>\n",
       "      <td>138902</td>\n",
       "      <td>mitochondrially encoded 12S RNA</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5948</th>\n",
       "      <td>Orphanet:90641</td>\n",
       "      <td>Mitochondrial non-syndromic sensorineural deaf...</td>\n",
       "      <td>17671</td>\n",
       "      <td>166766</td>\n",
       "      <td>mitochondrially encoded tRNA histidine</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TH</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5949</th>\n",
       "      <td>Orphanet:90641</td>\n",
       "      <td>Mitochondrial non-syndromic sensorineural deaf...</td>\n",
       "      <td>17722</td>\n",
       "      <td>167901</td>\n",
       "      <td>mitochondrially encoded tRNA serine 1 (UCN)</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TS1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6197</th>\n",
       "      <td>Orphanet:96191</td>\n",
       "      <td>Paternal uniparental disomy of chromosome 6</td>\n",
       "      <td>17383</td>\n",
       "      <td>159723</td>\n",
       "      <td>hydatidiform mole associated and imprinted (no...</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>HYMAI</td>\n",
       "      <td>Role in the phenotype of</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6199</th>\n",
       "      <td>Orphanet:96184</td>\n",
       "      <td>Maternal uniparental disomy of chromosome 14</td>\n",
       "      <td>17275</td>\n",
       "      <td>159158</td>\n",
       "      <td>maternally expressed 3 (non-protein coding)</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MEG3</td>\n",
       "      <td>Role in the phenotype of</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6205</th>\n",
       "      <td>Orphanet:96334</td>\n",
       "      <td>Paternal uniparental disomy of chromosome 14</td>\n",
       "      <td>17275</td>\n",
       "      <td>159158</td>\n",
       "      <td>maternally expressed 3 (non-protein coding)</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MEG3</td>\n",
       "      <td>Role in the phenotype of</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6407</th>\n",
       "      <td>Orphanet:99718</td>\n",
       "      <td>Leber plus disease</td>\n",
       "      <td>16480</td>\n",
       "      <td>123547</td>\n",
       "      <td>mitochondrially encoded NADH:ubiquinone oxidor...</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-ND3</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6408</th>\n",
       "      <td>Orphanet:99718</td>\n",
       "      <td>Leber plus disease</td>\n",
       "      <td>16481</td>\n",
       "      <td>123552</td>\n",
       "      <td>mitochondrially encoded NADH:ubiquinone oxidor...</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-ND4</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6409</th>\n",
       "      <td>Orphanet:99718</td>\n",
       "      <td>Leber plus disease</td>\n",
       "      <td>16484</td>\n",
       "      <td>123567</td>\n",
       "      <td>mitochondrially encoded NADH:ubiquinone oxidor...</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-ND6</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6453</th>\n",
       "      <td>Orphanet:98869</td>\n",
       "      <td>Congenital dyserythropoietic anemia type I</td>\n",
       "      <td>22273</td>\n",
       "      <td>362774</td>\n",
       "      <td>chromosome 15 open reading frame 41</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>C15ORF41</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6470</th>\n",
       "      <td>Orphanet:98890</td>\n",
       "      <td>Early-onset X-linked optic atrophy</td>\n",
       "      <td>21418</td>\n",
       "      <td>310868</td>\n",
       "      <td>optic atrophy 2 (obscure)</td>\n",
       "      <td>locus</td>\n",
       "      <td>OPA2</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6663</th>\n",
       "      <td>Orphanet:100973</td>\n",
       "      <td>FRAXE intellectual disability</td>\n",
       "      <td>17364</td>\n",
       "      <td>159590</td>\n",
       "      <td>fragile X mental retardation associated 3</td>\n",
       "      <td>locus</td>\n",
       "      <td>FMR3</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6675</th>\n",
       "      <td>Orphanet:100069</td>\n",
       "      <td>Semantic dementia</td>\n",
       "      <td>20676</td>\n",
       "      <td>286546</td>\n",
       "      <td>chromosome 9 open reading frame 72</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>C9ORF72</td>\n",
       "      <td>Major susceptibility factor in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6682</th>\n",
       "      <td>Orphanet:100070</td>\n",
       "      <td>Progressive non-fluent aphasia</td>\n",
       "      <td>20676</td>\n",
       "      <td>286546</td>\n",
       "      <td>chromosome 9 open reading frame 72</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>C9ORF72</td>\n",
       "      <td>Major susceptibility factor in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6691</th>\n",
       "      <td>Orphanet:100033</td>\n",
       "      <td>Hypomaturation amelogenesis imperfecta</td>\n",
       "      <td>21411</td>\n",
       "      <td>310402</td>\n",
       "      <td>chromosome 4 open reading frame 26</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>C4ORF26</td>\n",
       "      <td>Disease-causing germline mutation(s) (loss of ...</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6765</th>\n",
       "      <td>Orphanet:99845</td>\n",
       "      <td>Genetic recurrent myoglobinuria</td>\n",
       "      <td>16471</td>\n",
       "      <td>123512</td>\n",
       "      <td>mitochondrially encoded cytochrome c oxidase I</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6766</th>\n",
       "      <td>Orphanet:99845</td>\n",
       "      <td>Genetic recurrent myoglobinuria</td>\n",
       "      <td>16473</td>\n",
       "      <td>123520</td>\n",
       "      <td>mitochondrially encoded cytochrome c oxidase III</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-CO3</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6780</th>\n",
       "      <td>Orphanet:99852</td>\n",
       "      <td>Ravine syndrome</td>\n",
       "      <td>22738</td>\n",
       "      <td>398677</td>\n",
       "      <td>solute carrier family 7, member 2-intron1 (lon...</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>SLC7A2-IT1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>197 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        orphanet_code                                      orphanet_term  \\\n",
       "22          DOID:3687                                              MELAS   \n",
       "23          DOID:3687                                              MELAS   \n",
       "24          DOID:3687                                              MELAS   \n",
       "25          DOID:3687                                              MELAS   \n",
       "26          DOID:3687                                              MELAS   \n",
       "27          DOID:3687                                              MELAS   \n",
       "28          DOID:3687                                              MELAS   \n",
       "29          DOID:3687                                              MELAS   \n",
       "30          DOID:3687                                              MELAS   \n",
       "31          DOID:3687                                              MELAS   \n",
       "32          DOID:3687                                              MELAS   \n",
       "33          DOID:3687                                              MELAS   \n",
       "34          DOID:3687                                              MELAS   \n",
       "35          DOID:3687                                              MELAS   \n",
       "36         DOID:11727                      Facioscapulohumeral dystrophy   \n",
       "41         DOID:12934                              Kearns-Sayre syndrome   \n",
       "42         DOID:12934                              Kearns-Sayre syndrome   \n",
       "44           DOID:310                                              MERRF   \n",
       "45           DOID:310                                              MERRF   \n",
       "46           DOID:310                                              MERRF   \n",
       "47           DOID:310                                              MERRF   \n",
       "48           DOID:310                                              MERRF   \n",
       "49           DOID:310                                              MERRF   \n",
       "50           DOID:310                                              MERRF   \n",
       "51           DOID:310                                              MERRF   \n",
       "52           DOID:310                                              MERRF   \n",
       "53           DOID:310                                              MERRF   \n",
       "120          DOID:705                  Leber hereditary optic neuropathy   \n",
       "121          DOID:705                  Leber hereditary optic neuropathy   \n",
       "122          DOID:705                  Leber hereditary optic neuropathy   \n",
       "...               ...                                                ...   \n",
       "5054     Orphanet:225         Maternally-inherited diabetes and deafness   \n",
       "5108  Orphanet:401901  Huntington disease-like syndrome due to C9ORF7...   \n",
       "5178  Orphanet:391646                           Feingold syndrome type 2   \n",
       "5205  Orphanet:397959          TCR-alpha-beta-positive T-cell deficiency   \n",
       "5217  Orphanet:397750  Periodic paralysis with later-onset distal mot...   \n",
       "5218  Orphanet:397750  Periodic paralysis with later-onset distal mot...   \n",
       "5426  Orphanet:422526         Hereditary clear cell renal cell carcinoma   \n",
       "5456      Orphanet:88                         Idiopathic aplastic anemia   \n",
       "5570   Orphanet:65684                               Monomelic amyotrophy   \n",
       "5647   Orphanet:75327                   North Carolina macular dystrophy   \n",
       "5718   Orphanet:33110                       Autosomal agammaglobulinemia   \n",
       "5946   Orphanet:90641  Mitochondrial non-syndromic sensorineural deaf...   \n",
       "5947   Orphanet:90641  Mitochondrial non-syndromic sensorineural deaf...   \n",
       "5948   Orphanet:90641  Mitochondrial non-syndromic sensorineural deaf...   \n",
       "5949   Orphanet:90641  Mitochondrial non-syndromic sensorineural deaf...   \n",
       "6197   Orphanet:96191        Paternal uniparental disomy of chromosome 6   \n",
       "6199   Orphanet:96184       Maternal uniparental disomy of chromosome 14   \n",
       "6205   Orphanet:96334       Paternal uniparental disomy of chromosome 14   \n",
       "6407   Orphanet:99718                                 Leber plus disease   \n",
       "6408   Orphanet:99718                                 Leber plus disease   \n",
       "6409   Orphanet:99718                                 Leber plus disease   \n",
       "6453   Orphanet:98869         Congenital dyserythropoietic anemia type I   \n",
       "6470   Orphanet:98890                 Early-onset X-linked optic atrophy   \n",
       "6663  Orphanet:100973                      FRAXE intellectual disability   \n",
       "6675  Orphanet:100069                                  Semantic dementia   \n",
       "6682  Orphanet:100070                     Progressive non-fluent aphasia   \n",
       "6691  Orphanet:100033             Hypomaturation amelogenesis imperfecta   \n",
       "6765   Orphanet:99845                    Genetic recurrent myoglobinuria   \n",
       "6766   Orphanet:99845                    Genetic recurrent myoglobinuria   \n",
       "6780   Orphanet:99852                                    Ravine syndrome   \n",
       "\n",
       "      orphanet_gene_id  gene_orphanumber  \\\n",
       "22               16471            123512   \n",
       "23               16472            123516   \n",
       "24               16473            123520   \n",
       "25               16478            123537   \n",
       "26               16481            123552   \n",
       "27               16483            123562   \n",
       "28               16484            123567   \n",
       "29               16841            138895   \n",
       "30               16844            138906   \n",
       "31               17473            160307   \n",
       "32               17671            166766   \n",
       "33               17722            167901   \n",
       "34               17723            167905   \n",
       "35               17724            167909   \n",
       "36               15901            121239   \n",
       "41               16841            138895   \n",
       "42               17396            159773   \n",
       "44               16483            123562   \n",
       "45               16841            138895   \n",
       "46               16842            138900   \n",
       "47               16843            138902   \n",
       "48               16844            138906   \n",
       "49               17671            166766   \n",
       "50               17722            167901   \n",
       "51               17723            167905   \n",
       "52               17724            167909   \n",
       "53               18493            205934   \n",
       "120              16470            123508   \n",
       "121              16471            123512   \n",
       "122              16473            123520   \n",
       "...                ...               ...   \n",
       "5054             16981            140514   \n",
       "5108             20676            286546   \n",
       "5178             20656            285348   \n",
       "5205             22849            401908   \n",
       "5217             16470            123508   \n",
       "5218             17396            159773   \n",
       "5426             21868            326692   \n",
       "5456             16790            138457   \n",
       "5570             21089            299619   \n",
       "5647             23671            455932   \n",
       "5718             16247            122591   \n",
       "5946             16471            123512   \n",
       "5947             16843            138902   \n",
       "5948             17671            166766   \n",
       "5949             17722            167901   \n",
       "6197             17383            159723   \n",
       "6199             17275            159158   \n",
       "6205             17275            159158   \n",
       "6407             16480            123547   \n",
       "6408             16481            123552   \n",
       "6409             16484            123567   \n",
       "6453             22273            362774   \n",
       "6470             21418            310868   \n",
       "6663             17364            159590   \n",
       "6675             20676            286546   \n",
       "6682             20676            286546   \n",
       "6691             21411            310402   \n",
       "6765             16471            123512   \n",
       "6766             16473            123520   \n",
       "6780             22738            398677   \n",
       "\n",
       "                                              gene_name  \\\n",
       "22       mitochondrially encoded cytochrome c oxidase I   \n",
       "23      mitochondrially encoded cytochrome c oxidase II   \n",
       "24     mitochondrially encoded cytochrome c oxidase III   \n",
       "25    mitochondrially encoded NADH:ubiquinone oxidor...   \n",
       "26    mitochondrially encoded NADH:ubiquinone oxidor...   \n",
       "27    mitochondrially encoded NADH:ubiquinone oxidor...   \n",
       "28    mitochondrially encoded NADH:ubiquinone oxidor...   \n",
       "29       mitochondrially encoded tRNA leucine 1 (UUA/G)   \n",
       "30               mitochondrially encoded tRNA glutamine   \n",
       "31              mitochondrially encoded tRNA tryptophan   \n",
       "32               mitochondrially encoded tRNA histidine   \n",
       "33          mitochondrially encoded tRNA serine 1 (UCN)   \n",
       "34        mitochondrially encoded tRNA serine 2 (AGU/C)   \n",
       "35           mitochondrially encoded tRNA phenylalanine   \n",
       "36                             double homeobox 4 like 1   \n",
       "41       mitochondrially encoded tRNA leucine 1 (UUA/G)   \n",
       "42               mitochondrially encoded ATP synthase 8   \n",
       "44    mitochondrially encoded NADH:ubiquinone oxidor...   \n",
       "45       mitochondrially encoded tRNA leucine 1 (UUA/G)   \n",
       "46                  mitochondrially encoded tRNA lysine   \n",
       "47                      mitochondrially encoded 12S RNA   \n",
       "48               mitochondrially encoded tRNA glutamine   \n",
       "49               mitochondrially encoded tRNA histidine   \n",
       "50          mitochondrially encoded tRNA serine 1 (UCN)   \n",
       "51        mitochondrially encoded tRNA serine 2 (AGU/C)   \n",
       "52           mitochondrially encoded tRNA phenylalanine   \n",
       "53                 mitochondrially encoded tRNA proline   \n",
       "120              mitochondrially encoded ATP synthase 6   \n",
       "121      mitochondrially encoded cytochrome c oxidase I   \n",
       "122    mitochondrially encoded cytochrome c oxidase III   \n",
       "...                                                 ...   \n",
       "5054         mitochondrially encoded tRNA glutamic acid   \n",
       "5108                 chromosome 9 open reading frame 72   \n",
       "5178                     miR-17-92a-1 cluster host gene   \n",
       "5205                     T-cell receptor alpha constant   \n",
       "5217             mitochondrially encoded ATP synthase 6   \n",
       "5218             mitochondrially encoded ATP synthase 8   \n",
       "5426                     disrupted in renal carcinoma 3   \n",
       "5456                           telomerase RNA component   \n",
       "5570                 chromosome 5 open reading frame 42   \n",
       "5647      DNase1 hypersensitivity, chromosome 6, site 1   \n",
       "5718                   immunoglobulin heavy constant mu   \n",
       "5946     mitochondrially encoded cytochrome c oxidase I   \n",
       "5947                    mitochondrially encoded 12S RNA   \n",
       "5948             mitochondrially encoded tRNA histidine   \n",
       "5949        mitochondrially encoded tRNA serine 1 (UCN)   \n",
       "6197  hydatidiform mole associated and imprinted (no...   \n",
       "6199        maternally expressed 3 (non-protein coding)   \n",
       "6205        maternally expressed 3 (non-protein coding)   \n",
       "6407  mitochondrially encoded NADH:ubiquinone oxidor...   \n",
       "6408  mitochondrially encoded NADH:ubiquinone oxidor...   \n",
       "6409  mitochondrially encoded NADH:ubiquinone oxidor...   \n",
       "6453                chromosome 15 open reading frame 41   \n",
       "6470                          optic atrophy 2 (obscure)   \n",
       "6663          fragile X mental retardation associated 3   \n",
       "6675                 chromosome 9 open reading frame 72   \n",
       "6682                 chromosome 9 open reading frame 72   \n",
       "6691                 chromosome 4 open reading frame 26   \n",
       "6765     mitochondrially encoded cytochrome c oxidase I   \n",
       "6766   mitochondrially encoded cytochrome c oxidase III   \n",
       "6780  solute carrier family 7, member 2-intron1 (lon...   \n",
       "\n",
       "                      gene_type gene_symbol  \\\n",
       "22    gene with protein product      MT-CO1   \n",
       "23    gene with protein product      MT-CO2   \n",
       "24    gene with protein product      MT-CO3   \n",
       "25    gene with protein product      MT-ND1   \n",
       "26    gene with protein product      MT-ND4   \n",
       "27    gene with protein product      MT-ND5   \n",
       "28    gene with protein product      MT-ND6   \n",
       "29               Non-coding RNA      MT-TL1   \n",
       "30               Non-coding RNA       MT-TQ   \n",
       "31               Non-coding RNA       MT-TW   \n",
       "32               Non-coding RNA       MT-TH   \n",
       "33               Non-coding RNA      MT-TS1   \n",
       "34               Non-coding RNA      MT-TS2   \n",
       "35               Non-coding RNA       MT-TF   \n",
       "36    gene with protein product      DUX4L1   \n",
       "41               Non-coding RNA      MT-TL1   \n",
       "42    gene with protein product     MT-ATP8   \n",
       "44    gene with protein product      MT-ND5   \n",
       "45               Non-coding RNA      MT-TL1   \n",
       "46               Non-coding RNA       MT-TK   \n",
       "47               Non-coding RNA     MT-RNR1   \n",
       "48               Non-coding RNA       MT-TQ   \n",
       "49               Non-coding RNA       MT-TH   \n",
       "50               Non-coding RNA      MT-TS1   \n",
       "51               Non-coding RNA      MT-TS2   \n",
       "52               Non-coding RNA       MT-TF   \n",
       "53               Non-coding RNA       MT-TP   \n",
       "120   gene with protein product     MT-ATP6   \n",
       "121   gene with protein product      MT-CO1   \n",
       "122   gene with protein product      MT-CO3   \n",
       "...                         ...         ...   \n",
       "5054             Non-coding RNA       MT-TE   \n",
       "5108  gene with protein product     C9ORF72   \n",
       "5178             Non-coding RNA     MIR17HG   \n",
       "5205  gene with protein product        TRAC   \n",
       "5217  gene with protein product     MT-ATP6   \n",
       "5218  gene with protein product     MT-ATP8   \n",
       "5426  gene with protein product       DIRC3   \n",
       "5456             Non-coding RNA        TERC   \n",
       "5570  gene with protein product     C5ORF42   \n",
       "5647                      locus      DHS6S1   \n",
       "5718  gene with protein product        IGHM   \n",
       "5946  gene with protein product      MT-CO1   \n",
       "5947             Non-coding RNA     MT-RNR1   \n",
       "5948             Non-coding RNA       MT-TH   \n",
       "5949             Non-coding RNA      MT-TS1   \n",
       "6197             Non-coding RNA       HYMAI   \n",
       "6199             Non-coding RNA        MEG3   \n",
       "6205             Non-coding RNA        MEG3   \n",
       "6407  gene with protein product      MT-ND3   \n",
       "6408  gene with protein product      MT-ND4   \n",
       "6409  gene with protein product      MT-ND6   \n",
       "6453  gene with protein product    C15ORF41   \n",
       "6470                      locus        OPA2   \n",
       "6663                      locus        FMR3   \n",
       "6675  gene with protein product     C9ORF72   \n",
       "6682  gene with protein product     C9ORF72   \n",
       "6691  gene with protein product     C4ORF26   \n",
       "6765  gene with protein product      MT-CO1   \n",
       "6766  gene with protein product      MT-CO3   \n",
       "6780             Non-coding RNA  SLC7A2-IT1   \n",
       "\n",
       "                                               dga_type        dga_status  \\\n",
       "22              Disease-causing germline mutation(s) in          Assessed   \n",
       "23              Disease-causing germline mutation(s) in          Assessed   \n",
       "24                             Candidate gene tested in  Not yet assessed   \n",
       "25              Disease-causing germline mutation(s) in          Assessed   \n",
       "26              Disease-causing germline mutation(s) in          Assessed   \n",
       "27              Disease-causing germline mutation(s) in          Assessed   \n",
       "28              Disease-causing germline mutation(s) in          Assessed   \n",
       "29              Disease-causing germline mutation(s) in          Assessed   \n",
       "30                             Candidate gene tested in  Not yet assessed   \n",
       "31              Disease-causing germline mutation(s) in          Assessed   \n",
       "32                             Candidate gene tested in  Not yet assessed   \n",
       "33              Disease-causing germline mutation(s) in          Assessed   \n",
       "34                             Candidate gene tested in  Not yet assessed   \n",
       "35              Disease-causing germline mutation(s) in          Assessed   \n",
       "36                             Candidate gene tested in  Not yet assessed   \n",
       "41                             Candidate gene tested in  Not yet assessed   \n",
       "42                             Candidate gene tested in  Not yet assessed   \n",
       "44              Disease-causing germline mutation(s) in          Assessed   \n",
       "45              Disease-causing germline mutation(s) in          Assessed   \n",
       "46              Disease-causing germline mutation(s) in          Assessed   \n",
       "47                             Candidate gene tested in  Not yet assessed   \n",
       "48                             Candidate gene tested in  Not yet assessed   \n",
       "49              Disease-causing germline mutation(s) in          Assessed   \n",
       "50              Disease-causing germline mutation(s) in          Assessed   \n",
       "51                             Candidate gene tested in  Not yet assessed   \n",
       "52              Disease-causing germline mutation(s) in          Assessed   \n",
       "53              Disease-causing germline mutation(s) in          Assessed   \n",
       "120             Disease-causing germline mutation(s) in          Assessed   \n",
       "121                            Candidate gene tested in  Not yet assessed   \n",
       "122             Disease-causing germline mutation(s) in          Assessed   \n",
       "...                                                 ...               ...   \n",
       "5054            Disease-causing germline mutation(s) in          Assessed   \n",
       "5108            Disease-causing germline mutation(s) in          Assessed   \n",
       "5178                           Role in the phenotype of          Assessed   \n",
       "5205            Disease-causing germline mutation(s) in          Assessed   \n",
       "5217            Disease-causing germline mutation(s) in          Assessed   \n",
       "5218            Disease-causing germline mutation(s) in          Assessed   \n",
       "5426                           Part of a fusion gene in          Assessed   \n",
       "5456                     Major susceptibility factor in          Assessed   \n",
       "5570                     Major susceptibility factor in          Assessed   \n",
       "5647            Disease-causing germline mutation(s) in          Assessed   \n",
       "5718            Disease-causing germline mutation(s) in          Assessed   \n",
       "5946            Disease-causing germline mutation(s) in          Assessed   \n",
       "5947            Disease-causing germline mutation(s) in          Assessed   \n",
       "5948            Disease-causing germline mutation(s) in          Assessed   \n",
       "5949            Disease-causing germline mutation(s) in          Assessed   \n",
       "6197                           Role in the phenotype of  Not yet assessed   \n",
       "6199                           Role in the phenotype of          Assessed   \n",
       "6205                           Role in the phenotype of          Assessed   \n",
       "6407            Disease-causing germline mutation(s) in  Not yet assessed   \n",
       "6408            Disease-causing germline mutation(s) in  Not yet assessed   \n",
       "6409            Disease-causing germline mutation(s) in          Assessed   \n",
       "6453            Disease-causing germline mutation(s) in          Assessed   \n",
       "6470            Disease-causing germline mutation(s) in          Assessed   \n",
       "6663            Disease-causing germline mutation(s) in          Assessed   \n",
       "6675                     Major susceptibility factor in  Not yet assessed   \n",
       "6682                     Major susceptibility factor in  Not yet assessed   \n",
       "6691  Disease-causing germline mutation(s) (loss of ...          Assessed   \n",
       "6765            Disease-causing germline mutation(s) in          Assessed   \n",
       "6766            Disease-causing germline mutation(s) in          Assessed   \n",
       "6780            Disease-causing germline mutation(s) in          Assessed   \n",
       "\n",
       "     entrez_gene_id  \n",
       "22          missing  \n",
       "23          missing  \n",
       "24          missing  \n",
       "25          missing  \n",
       "26          missing  \n",
       "27          missing  \n",
       "28          missing  \n",
       "29          missing  \n",
       "30          missing  \n",
       "31          missing  \n",
       "32          missing  \n",
       "33          missing  \n",
       "34          missing  \n",
       "35          missing  \n",
       "36          missing  \n",
       "41          missing  \n",
       "42          missing  \n",
       "44          missing  \n",
       "45          missing  \n",
       "46          missing  \n",
       "47          missing  \n",
       "48          missing  \n",
       "49          missing  \n",
       "50          missing  \n",
       "51          missing  \n",
       "52          missing  \n",
       "53          missing  \n",
       "120         missing  \n",
       "121         missing  \n",
       "122         missing  \n",
       "...             ...  \n",
       "5054        missing  \n",
       "5108        missing  \n",
       "5178        missing  \n",
       "5205        missing  \n",
       "5217        missing  \n",
       "5218        missing  \n",
       "5426        missing  \n",
       "5456        missing  \n",
       "5570        missing  \n",
       "5647        missing  \n",
       "5718        missing  \n",
       "5946        missing  \n",
       "5947        missing  \n",
       "5948        missing  \n",
       "5949        missing  \n",
       "6197        missing  \n",
       "6199        missing  \n",
       "6205        missing  \n",
       "6407        missing  \n",
       "6408        missing  \n",
       "6409        missing  \n",
       "6453        missing  \n",
       "6470        missing  \n",
       "6663        missing  \n",
       "6675        missing  \n",
       "6682        missing  \n",
       "6691        missing  \n",
       "6765        missing  \n",
       "6766        missing  \n",
       "6780        missing  \n",
       "\n",
       "[197 rows x 10 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check if there are missing values\n",
    "orpha_na = orphanet_df.fillna('missing')\n",
    "orpha_na.query('entrez_gene_id == \"missing\"')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'NA GeneID: 197'"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'NA GeneID: {}'.format(len(orpha_na.query('entrez_gene_id == \"missing\"')))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'NA GeneID for protein codeing: 88'"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'NA GeneID for protein codeing: {}'.format(len(orpha_na.query('entrez_gene_id == \"missing\" and gene_type == \"gene with protein product\"')))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orphanet_code</th>\n",
       "      <th>orphanet_term</th>\n",
       "      <th>orphanet_gene_id</th>\n",
       "      <th>gene_orphanumber</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>dga_type</th>\n",
       "      <th>dga_status</th>\n",
       "      <th>entrez_gene_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>DOID:3687</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>16471</td>\n",
       "      <td>123512</td>\n",
       "      <td>mitochondrially encoded cytochrome c oxidase I</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>DOID:3687</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>16472</td>\n",
       "      <td>123516</td>\n",
       "      <td>mitochondrially encoded cytochrome c oxidase II</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-CO2</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   orphanet_code orphanet_term  orphanet_gene_id  gene_orphanumber  \\\n",
       "22     DOID:3687         MELAS             16471            123512   \n",
       "23     DOID:3687         MELAS             16472            123516   \n",
       "\n",
       "                                          gene_name  \\\n",
       "22   mitochondrially encoded cytochrome c oxidase I   \n",
       "23  mitochondrially encoded cytochrome c oxidase II   \n",
       "\n",
       "                    gene_type gene_symbol  \\\n",
       "22  gene with protein product      MT-CO1   \n",
       "23  gene with protein product      MT-CO2   \n",
       "\n",
       "                                   dga_type dga_status entrez_gene_id  \n",
       "22  Disease-causing germline mutation(s) in   Assessed        missing  \n",
       "23  Disease-causing germline mutation(s) in   Assessed        missing  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orpha_na.query('entrez_gene_id == \"missing\" and gene_type == \"gene with protein product\"').head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>entrez_gene_id</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [entrez_gene_id, gene_symbol]\n",
       "Index: []"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_df.query('gene_symbol == \"MT-CO1\"')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>entrez_gene_id</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3622</th>\n",
       "      <td>4512</td>\n",
       "      <td>COX1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      entrez_gene_id gene_symbol\n",
       "3622            4512        COX1"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_df.query('entrez_gene_id == 4512')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# the problem is that in the web site the officia symbol for entrez:4512 is 'MT-CO1',\n",
    "# while in the downloaded file is 'COX1' +> try the mapping using biothings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "genes pre-removal: 6806\n",
      "genes post-removal: 6609\n"
     ]
    }
   ],
   "source": [
    "# Remove genes with missing entrez_gene_id data\n",
    "orpha_df = orpha_na.query('entrez_gene_id != \"missing\"')\n",
    "print('genes pre-removal: {}'.format(len(orpha_na)))\n",
    "print('genes post-removal: {}'.format(len(orpha_df)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>doid_id</th>\n",
       "      <th>doid_name</th>\n",
       "      <th>orphanet_gene_id</th>\n",
       "      <th>gene_orphanumber</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>dga_type</th>\n",
       "      <th>dga_status</th>\n",
       "      <th>entrez_gene_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DOID:0050461</td>\n",
       "      <td>Aspartylglucosaminuria</td>\n",
       "      <td>15470</td>\n",
       "      <td>119513</td>\n",
       "      <td>aspartylglucosaminidase</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>AGA</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DOID:0050441</td>\n",
       "      <td>Multiple sulfatase deficiency</td>\n",
       "      <td>15570</td>\n",
       "      <td>119899</td>\n",
       "      <td>sulfatase modifying factor 1</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>SUMF1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>285362</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        doid_id                      doid_name  orphanet_gene_id  \\\n",
       "0  DOID:0050461         Aspartylglucosaminuria             15470   \n",
       "1  DOID:0050441  Multiple sulfatase deficiency             15570   \n",
       "\n",
       "   gene_orphanumber                     gene_name                  gene_type  \\\n",
       "0            119513       aspartylglucosaminidase  gene with protein product   \n",
       "1            119899  sulfatase modifying factor 1  gene with protein product   \n",
       "\n",
       "  gene_symbol                                 dga_type dga_status  \\\n",
       "0         AGA  Disease-causing germline mutation(s) in   Assessed   \n",
       "1       SUMF1  Disease-causing germline mutation(s) in   Assessed   \n",
       "\n",
       "  entrez_gene_id  \n",
       "0            175  \n",
       "1         285362  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# reformat\n",
    "orpha_df = orpha_df.rename(columns={'orphanet_code': 'doid_id', 'orphanet_term': 'doid_name'})\n",
    "#orpha_df['doid_id'] = 'ORPHA:' + orpha_df['doid_id'].astype(str)\n",
    "orpha_df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>doid_id</th>\n",
       "      <th>doid_name</th>\n",
       "      <th>entrez_gene_id</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>score_text</th>\n",
       "      <th>score_knowledge</th>\n",
       "      <th>score_cosmic</th>\n",
       "      <th>score_distild</th>\n",
       "      <th>score_integrated_no_distild</th>\n",
       "      <th>score_integrated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DOID:13223</td>\n",
       "      <td>uterine fibroid</td>\n",
       "      <td>60</td>\n",
       "      <td>ACTB</td>\n",
       "      <td>0.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DOID:13223</td>\n",
       "      <td>uterine fibroid</td>\n",
       "      <td>71</td>\n",
       "      <td>ACTG1</td>\n",
       "      <td>0.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      doid_id        doid_name  entrez_gene_id gene_symbol  score_text  \\\n",
       "0  DOID:13223  uterine fibroid              60        ACTB         0.8   \n",
       "1  DOID:13223  uterine fibroid              71       ACTG1         0.8   \n",
       "\n",
       "   score_knowledge  score_cosmic  score_distild  score_integrated_no_distild  \\\n",
       "0              NaN           NaN            NaN                          0.8   \n",
       "1              NaN           NaN            NaN                          0.8   \n",
       "\n",
       "   score_integrated  \n",
       "0               0.8  \n",
       "1               0.8  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# DISEASES\n",
    "commit = 'e0089ef89a56348d7d4e0684a9c51c5747b16237'\n",
    "#url = utils.rawgit('dhimmel', 'diseases', commit, 'data/merged-slim.tsv')\n",
    "url = rawgit + 'dhimmel/' + 'diseases/' + commit + '/data/merged-slim.tsv'\n",
    "diseases_df = pandas.read_table(url)\n",
    "diseases_df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>doid_id</th>\n",
       "      <th>doid_name</th>\n",
       "      <th>entrez_gene_id</th>\n",
       "      <th>Symbol</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DOID:0001816</td>\n",
       "      <td>angiosarcoma</td>\n",
       "      <td>302</td>\n",
       "      <td>ANXA2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DOID:0001816</td>\n",
       "      <td>angiosarcoma</td>\n",
       "      <td>595</td>\n",
       "      <td>CCND1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        doid_id     doid_name  entrez_gene_id Symbol  count\n",
       "0  DOID:0001816  angiosarcoma             302  ANXA2      1\n",
       "1  DOID:0001816  angiosarcoma             595  CCND1      1"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# DOAF\n",
    "commit = 'bbe1c326aa385416e36d02b144e89e2b99e700b6'\n",
    "#url = utils.rawgit('dhimmel', 'doaf', commit, 'data/doaf.tsv')\n",
    "url = rawgit + 'dhimmel/' + 'doaf/' + commit + '/data/doaf.tsv'\n",
    "doaf_df = pandas.read_table(url)\n",
    "doaf_df = doaf_df.rename(columns={'doid_code': 'doid_id', 'GeneID': 'entrez_gene_id'})\n",
    "doaf_df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>doid_id</th>\n",
       "      <th>doid_name</th>\n",
       "      <th>entrez_gene_id</th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>count</th>\n",
       "      <th>pubmeds_max</th>\n",
       "      <th>score_max</th>\n",
       "      <th>score_mean</th>\n",
       "      <th>associationType</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>idiopathic pulmonary fibrosis</td>\n",
       "      <td>729238</td>\n",
       "      <td>SFTPA2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.620284</td>\n",
       "      <td>0.620284</td>\n",
       "      <td>Biomarker|GeneticVariation</td>\n",
       "      <td>BeFree|CLINVAR|CTD_human|UNIPROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>idiopathic pulmonary fibrosis</td>\n",
       "      <td>7015</td>\n",
       "      <td>TERT</td>\n",
       "      <td>1.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.422153</td>\n",
       "      <td>0.422153</td>\n",
       "      <td>Biomarker|GeneticVariation</td>\n",
       "      <td>BeFree|CLINVAR|CTD_human|GAD|LHGDN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        doid_id                      doid_name  entrez_gene_id geneSymbol  \\\n",
       "0  DOID:0050156  idiopathic pulmonary fibrosis          729238     SFTPA2   \n",
       "1  DOID:0050156  idiopathic pulmonary fibrosis            7015       TERT   \n",
       "\n",
       "   count  pubmeds_max  score_max  score_mean             associationType  \\\n",
       "0    1.0          1.0   0.620284    0.620284  Biomarker|GeneticVariation   \n",
       "1    1.0         10.0   0.422153    0.422153  Biomarker|GeneticVariation   \n",
       "\n",
       "                               source  \n",
       "0    BeFree|CLINVAR|CTD_human|UNIPROT  \n",
       "1  BeFree|CLINVAR|CTD_human|GAD|LHGDN  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# DisGeNET\n",
    "commit = 'fdc5f42f2da745cbf71d7b4cc5021de5685e4a11'\n",
    "#url = utils.rawgit('dhimmel', 'disgenet', commit, 'data/consolidated.tsv')\n",
    "url = rawgit + 'dhimmel/' + 'disgenet/' + commit + '/data/consolidated.tsv'\n",
    "disgenet_df = pandas.read_table(url)\n",
    "disgenet_df = disgenet_df.rename(columns={'doid_code': 'doid_id', 'geneId': 'entrez_gene_id'})\n",
    "disgenet_df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>doid_id</th>\n",
       "      <th>doid_name</th>\n",
       "      <th>locus</th>\n",
       "      <th>high_confidence</th>\n",
       "      <th>primary</th>\n",
       "      <th>status</th>\n",
       "      <th>entrez_gene_id</th>\n",
       "      <th>symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DOID:9970</td>\n",
       "      <td>obesity</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>HC-P</td>\n",
       "      <td>3953</td>\n",
       "      <td>LEPR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DOID:9970</td>\n",
       "      <td>obesity</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>HC-P</td>\n",
       "      <td>4094</td>\n",
       "      <td>MAF</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     doid_id doid_name  locus  high_confidence  primary status  \\\n",
       "0  DOID:9970   obesity      0                1        1   HC-P   \n",
       "1  DOID:9970   obesity     14                1        1   HC-P   \n",
       "\n",
       "   entrez_gene_id symbol  \n",
       "0            3953   LEPR  \n",
       "1            4094    MAF  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# hetio GWAS\n",
    "commit = '0617ea7ea8268f21f5ca1b8dbe487dd12671fc7b'\n",
    "#url = utils.rawgit('dhimmel', 'gwas-catalog', commit, 'data/gene-associations.tsv')\n",
    "url = rawgit + 'dhimmel/' + 'gwas-catalog/' + commit + '/data/gene-associations.tsv'\n",
    "gwas_df = pandas.read_table(url)\n",
    "gwas_df = gwas_df.rename(columns={'doid_code': 'doid_id', 'gene': 'entrez_gene_id'})\n",
    "gwas_df.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Filters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "orpha_df = orpha_df.query('gene_type == \"gene with protein product\" or dga_status == \"Assessed\"')\n",
    "diseases_df = diseases_df.query('score_integrated_no_distild >= 2')\n",
    "doaf_df = doaf_df.query('count >= 3')\n",
    "disgenet_df = disgenet_df.query('score_max >= 0.06')\n",
    "gwas_df = gwas_df[gwas_df.status == 'HC-P']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Combine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "orpha_df['provenance'] = 'Orphanet'\n",
    "diseases_df['provenance'] = 'DISEASES'\n",
    "doaf_df['provenance'] = 'DOAF'\n",
    "disgenet_df['provenance'] = 'DisGeNET'\n",
    "gwas_df['provenance'] = 'GWAS Catalog'\n",
    "\n",
    "orpha_df['license'] = 'CC BY-ND 3.0'\n",
    "diseases_df['license'] = 'CC BY 4.0'\n",
    "doaf_df['license'] = ''\n",
    "disgenet_df['license'] = 'ODbL 1.0'\n",
    "gwas_df['license'] = 'CC BY 4.0'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "DisGeNET        7552\n",
       "DISEASES        4990\n",
       "DOAF            1649\n",
       "GWAS Catalog    1283\n",
       "Name: provenance, dtype: int64"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs = [df[['doid_id', 'entrez_gene_id', 'provenance', 'license']]\n",
    "       for df in (diseases_df, doaf_df, disgenet_df, gwas_df)]\n",
    "concat_df = pandas.concat(dfs)\n",
    "concat_df = disease_df.merge(gene_df.merge(concat_df))\n",
    "concat_df.provenance.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "DisGeNET        7552\n",
       "Orphanet        6609\n",
       "DISEASES        4990\n",
       "DOAF            1649\n",
       "GWAS Catalog    1283\n",
       "Name: provenance, dtype: int64"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concat_df.head(2)\n",
    "orpha_df = orpha_df[['doid_id', 'doid_name', 'entrez_gene_id', 'gene_symbol', 'provenance', 'license']]\n",
    "concat_df = pandas.concat([concat_df,orpha_df])\n",
    "concat_df.provenance.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>doid_id</th>\n",
       "      <th>doid_name</th>\n",
       "      <th>entrez_gene_id</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>sources</th>\n",
       "      <th>license</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DOID:2531</td>\n",
       "      <td>Hematologic cancer</td>\n",
       "      <td>25</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>DISEASES|DisGeNET</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DOID:2531</td>\n",
       "      <td>Hematologic cancer</td>\n",
       "      <td>27</td>\n",
       "      <td>ABL2</td>\n",
       "      <td>DisGeNET</td>\n",
       "      <td>ODbL 1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DOID:2531</td>\n",
       "      <td>Hematologic cancer</td>\n",
       "      <td>54</td>\n",
       "      <td>ACP5</td>\n",
       "      <td>DISEASES</td>\n",
       "      <td>CC BY 4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DOID:2531</td>\n",
       "      <td>Hematologic cancer</td>\n",
       "      <td>113</td>\n",
       "      <td>ADCY7</td>\n",
       "      <td>DisGeNET</td>\n",
       "      <td>ODbL 1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>DOID:2531</td>\n",
       "      <td>Hematologic cancer</td>\n",
       "      <td>142</td>\n",
       "      <td>PARP1</td>\n",
       "      <td>DISEASES|DisGeNET</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     doid_id           doid_name  entrez_gene_id gene_symbol  \\\n",
       "0  DOID:2531  Hematologic cancer              25        ABL1   \n",
       "1  DOID:2531  Hematologic cancer              27        ABL2   \n",
       "2  DOID:2531  Hematologic cancer              54        ACP5   \n",
       "3  DOID:2531  Hematologic cancer             113       ADCY7   \n",
       "4  DOID:2531  Hematologic cancer             142       PARP1   \n",
       "\n",
       "             sources    license  \n",
       "0  DISEASES|DisGeNET       None  \n",
       "1           DisGeNET   ODbL 1.0  \n",
       "2           DISEASES  CC BY 4.0  \n",
       "3           DisGeNET   ODbL 1.0  \n",
       "4  DISEASES|DisGeNET       None  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def condense(df):\n",
    "    \"\"\"Consolidate multiple associations into a single Series.\"\"\"\n",
    "    row = pandas.Series()\n",
    "    row['sources'] = '|'.join(df.provenance)\n",
    "    licenses = set(df.license)\n",
    "    licenses.discard('')\n",
    "    try:\n",
    "        row['license'], = licenses\n",
    "    except ValueError:\n",
    "        row['license'] = None\n",
    "    return row\n",
    "\n",
    "short_df = concat_df.groupby(['doid_id', 'entrez_gene_id']).apply(condense).reset_index()\n",
    "short_df = disease_df.merge(gene_df.merge(short_df))\n",
    "short_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "short_df.to_csv('DaG-association.tsv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "short_df = concat_df.groupby(['doid_id', 'entrez_gene_id']).apply(condense).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>doid_id</th>\n",
       "      <th>entrez_gene_id</th>\n",
       "      <th>sources</th>\n",
       "      <th>license</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>1832.0</td>\n",
       "      <td>DisGeNET|Orphanet</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>2335.0</td>\n",
       "      <td>DISEASES</td>\n",
       "      <td>CC BY 4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>2355.0</td>\n",
       "      <td>DisGeNET</td>\n",
       "      <td>ODbL 1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>4088.0</td>\n",
       "      <td>DISEASES</td>\n",
       "      <td>CC BY 4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>5073.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>5328.0</td>\n",
       "      <td>DisGeNET</td>\n",
       "      <td>ODbL 1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>6440.0</td>\n",
       "      <td>DISEASES|DisGeNET|Orphanet</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>6441.0</td>\n",
       "      <td>DISEASES</td>\n",
       "      <td>CC BY 4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>7015.0</td>\n",
       "      <td>DISEASES|DOAF|DisGeNET|GWAS Catalog|Orphanet</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>7040.0</td>\n",
       "      <td>DISEASES</td>\n",
       "      <td>CC BY 4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>10144.0</td>\n",
       "      <td>DisGeNET|Orphanet</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>23250.0</td>\n",
       "      <td>DisGeNET|Orphanet</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>51750.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>54472.0</td>\n",
       "      <td>GWAS Catalog</td>\n",
       "      <td>CC BY 4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>79991.0</td>\n",
       "      <td>DisGeNET|Orphanet</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>91039.0</td>\n",
       "      <td>DisGeNET|Orphanet</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>255520.0</td>\n",
       "      <td>DISEASES|DisGeNET</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>653509.0</td>\n",
       "      <td>DisGeNET|Orphanet</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>727897.0</td>\n",
       "      <td>DisGeNET|Orphanet</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>729238.0</td>\n",
       "      <td>DISEASES|DisGeNET|Orphanet</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>DOID:0050167</td>\n",
       "      <td>326.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>DOID:0050331</td>\n",
       "      <td>2255.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>DOID:0050331</td>\n",
       "      <td>2261.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>DOID:0050331</td>\n",
       "      <td>2263.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>DOID:0050335</td>\n",
       "      <td>8787.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>DOID:0050335</td>\n",
       "      <td>388531.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>DOID:0050425</td>\n",
       "      <td>1775.0</td>\n",
       "      <td>DISEASES</td>\n",
       "      <td>CC BY 4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>DOID:0050425</td>\n",
       "      <td>4211.0</td>\n",
       "      <td>DISEASES|DisGeNET|GWAS Catalog</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>DOID:0050425</td>\n",
       "      <td>5348.0</td>\n",
       "      <td>DISEASES</td>\n",
       "      <td>CC BY 4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>DOID:0050425</td>\n",
       "      <td>5443.0</td>\n",
       "      <td>DisGeNET</td>\n",
       "      <td>ODbL 1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19143</th>\n",
       "      <td>Orphanet:99861</td>\n",
       "      <td>6965.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19144</th>\n",
       "      <td>Orphanet:99861</td>\n",
       "      <td>8028.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19145</th>\n",
       "      <td>Orphanet:99861</td>\n",
       "      <td>8115.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19146</th>\n",
       "      <td>Orphanet:99861</td>\n",
       "      <td>8301.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19147</th>\n",
       "      <td>Orphanet:99861</td>\n",
       "      <td>30012.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19148</th>\n",
       "      <td>Orphanet:99872</td>\n",
       "      <td>673.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19149</th>\n",
       "      <td>Orphanet:99875</td>\n",
       "      <td>1277.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19150</th>\n",
       "      <td>Orphanet:99876</td>\n",
       "      <td>1278.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19151</th>\n",
       "      <td>Orphanet:99879</td>\n",
       "      <td>4221.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19152</th>\n",
       "      <td>Orphanet:99879</td>\n",
       "      <td>79577.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19153</th>\n",
       "      <td>Orphanet:99880</td>\n",
       "      <td>79577.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19154</th>\n",
       "      <td>Orphanet:99887</td>\n",
       "      <td>2623.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19155</th>\n",
       "      <td>Orphanet:99898</td>\n",
       "      <td>3459.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19156</th>\n",
       "      <td>Orphanet:99900</td>\n",
       "      <td>33.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19157</th>\n",
       "      <td>Orphanet:99901</td>\n",
       "      <td>28976.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19158</th>\n",
       "      <td>Orphanet:99914</td>\n",
       "      <td>23405.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19159</th>\n",
       "      <td>Orphanet:99916</td>\n",
       "      <td>23405.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19160</th>\n",
       "      <td>Orphanet:99944</td>\n",
       "      <td>54332.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19161</th>\n",
       "      <td>Orphanet:99960</td>\n",
       "      <td>5205.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19162</th>\n",
       "      <td>Orphanet:99961</td>\n",
       "      <td>8647.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19163</th>\n",
       "      <td>Orphanet:99966</td>\n",
       "      <td>6598.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19164</th>\n",
       "      <td>Orphanet:99967</td>\n",
       "      <td>1649.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19165</th>\n",
       "      <td>Orphanet:99967</td>\n",
       "      <td>2521.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19166</th>\n",
       "      <td>Orphanet:99970</td>\n",
       "      <td>1019.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19167</th>\n",
       "      <td>Orphanet:99970</td>\n",
       "      <td>4193.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19168</th>\n",
       "      <td>Orphanet:99970</td>\n",
       "      <td>8091.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19169</th>\n",
       "      <td>Orphanet:99971</td>\n",
       "      <td>1019.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19170</th>\n",
       "      <td>Orphanet:99971</td>\n",
       "      <td>4193.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19171</th>\n",
       "      <td>Orphanet:99971</td>\n",
       "      <td>8091.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19172</th>\n",
       "      <td>Orphanet:99989</td>\n",
       "      <td>3767.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19173 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              doid_id  entrez_gene_id  \\\n",
       "0        DOID:0050156          1832.0   \n",
       "1        DOID:0050156          2335.0   \n",
       "2        DOID:0050156          2355.0   \n",
       "3        DOID:0050156          4088.0   \n",
       "4        DOID:0050156          5073.0   \n",
       "5        DOID:0050156          5328.0   \n",
       "6        DOID:0050156          6440.0   \n",
       "7        DOID:0050156          6441.0   \n",
       "8        DOID:0050156          7015.0   \n",
       "9        DOID:0050156          7040.0   \n",
       "10       DOID:0050156         10144.0   \n",
       "11       DOID:0050156         23250.0   \n",
       "12       DOID:0050156         51750.0   \n",
       "13       DOID:0050156         54472.0   \n",
       "14       DOID:0050156         79991.0   \n",
       "15       DOID:0050156         91039.0   \n",
       "16       DOID:0050156        255520.0   \n",
       "17       DOID:0050156        653509.0   \n",
       "18       DOID:0050156        727897.0   \n",
       "19       DOID:0050156        729238.0   \n",
       "20       DOID:0050167           326.0   \n",
       "21       DOID:0050331          2255.0   \n",
       "22       DOID:0050331          2261.0   \n",
       "23       DOID:0050331          2263.0   \n",
       "24       DOID:0050335          8787.0   \n",
       "25       DOID:0050335        388531.0   \n",
       "26       DOID:0050425          1775.0   \n",
       "27       DOID:0050425          4211.0   \n",
       "28       DOID:0050425          5348.0   \n",
       "29       DOID:0050425          5443.0   \n",
       "...               ...             ...   \n",
       "19143  Orphanet:99861          6965.0   \n",
       "19144  Orphanet:99861          8028.0   \n",
       "19145  Orphanet:99861          8115.0   \n",
       "19146  Orphanet:99861          8301.0   \n",
       "19147  Orphanet:99861         30012.0   \n",
       "19148  Orphanet:99872           673.0   \n",
       "19149  Orphanet:99875          1277.0   \n",
       "19150  Orphanet:99876          1278.0   \n",
       "19151  Orphanet:99879          4221.0   \n",
       "19152  Orphanet:99879         79577.0   \n",
       "19153  Orphanet:99880         79577.0   \n",
       "19154  Orphanet:99887          2623.0   \n",
       "19155  Orphanet:99898          3459.0   \n",
       "19156  Orphanet:99900            33.0   \n",
       "19157  Orphanet:99901         28976.0   \n",
       "19158  Orphanet:99914         23405.0   \n",
       "19159  Orphanet:99916         23405.0   \n",
       "19160  Orphanet:99944         54332.0   \n",
       "19161  Orphanet:99960          5205.0   \n",
       "19162  Orphanet:99961          8647.0   \n",
       "19163  Orphanet:99966          6598.0   \n",
       "19164  Orphanet:99967          1649.0   \n",
       "19165  Orphanet:99967          2521.0   \n",
       "19166  Orphanet:99970          1019.0   \n",
       "19167  Orphanet:99970          4193.0   \n",
       "19168  Orphanet:99970          8091.0   \n",
       "19169  Orphanet:99971          1019.0   \n",
       "19170  Orphanet:99971          4193.0   \n",
       "19171  Orphanet:99971          8091.0   \n",
       "19172  Orphanet:99989          3767.0   \n",
       "\n",
       "                                            sources       license  \n",
       "0                                 DisGeNET|Orphanet          None  \n",
       "1                                          DISEASES     CC BY 4.0  \n",
       "2                                          DisGeNET      ODbL 1.0  \n",
       "3                                          DISEASES     CC BY 4.0  \n",
       "4                                          Orphanet  CC BY-ND 3.0  \n",
       "5                                          DisGeNET      ODbL 1.0  \n",
       "6                        DISEASES|DisGeNET|Orphanet          None  \n",
       "7                                          DISEASES     CC BY 4.0  \n",
       "8      DISEASES|DOAF|DisGeNET|GWAS Catalog|Orphanet          None  \n",
       "9                                          DISEASES     CC BY 4.0  \n",
       "10                                DisGeNET|Orphanet          None  \n",
       "11                                DisGeNET|Orphanet          None  \n",
       "12                                         Orphanet  CC BY-ND 3.0  \n",
       "13                                     GWAS Catalog     CC BY 4.0  \n",
       "14                                DisGeNET|Orphanet          None  \n",
       "15                                DisGeNET|Orphanet          None  \n",
       "16                                DISEASES|DisGeNET          None  \n",
       "17                                DisGeNET|Orphanet          None  \n",
       "18                                DisGeNET|Orphanet          None  \n",
       "19                       DISEASES|DisGeNET|Orphanet          None  \n",
       "20                                         Orphanet  CC BY-ND 3.0  \n",
       "21                                         Orphanet  CC BY-ND 3.0  \n",
       "22                                         Orphanet  CC BY-ND 3.0  \n",
       "23                                         Orphanet  CC BY-ND 3.0  \n",
       "24                                         Orphanet  CC BY-ND 3.0  \n",
       "25                                         Orphanet  CC BY-ND 3.0  \n",
       "26                                         DISEASES     CC BY 4.0  \n",
       "27                   DISEASES|DisGeNET|GWAS Catalog          None  \n",
       "28                                         DISEASES     CC BY 4.0  \n",
       "29                                         DisGeNET      ODbL 1.0  \n",
       "...                                             ...           ...  \n",
       "19143                                      Orphanet  CC BY-ND 3.0  \n",
       "19144                                      Orphanet  CC BY-ND 3.0  \n",
       "19145                                      Orphanet  CC BY-ND 3.0  \n",
       "19146                                      Orphanet  CC BY-ND 3.0  \n",
       "19147                                      Orphanet  CC BY-ND 3.0  \n",
       "19148                                      Orphanet  CC BY-ND 3.0  \n",
       "19149                                      Orphanet  CC BY-ND 3.0  \n",
       "19150                                      Orphanet  CC BY-ND 3.0  \n",
       "19151                                      Orphanet  CC BY-ND 3.0  \n",
       "19152                                      Orphanet  CC BY-ND 3.0  \n",
       "19153                                      Orphanet  CC BY-ND 3.0  \n",
       "19154                                      Orphanet  CC BY-ND 3.0  \n",
       "19155                                      Orphanet  CC BY-ND 3.0  \n",
       "19156                                      Orphanet  CC BY-ND 3.0  \n",
       "19157                                      Orphanet  CC BY-ND 3.0  \n",
       "19158                                      Orphanet  CC BY-ND 3.0  \n",
       "19159                                      Orphanet  CC BY-ND 3.0  \n",
       "19160                                      Orphanet  CC BY-ND 3.0  \n",
       "19161                                      Orphanet  CC BY-ND 3.0  \n",
       "19162                                      Orphanet  CC BY-ND 3.0  \n",
       "19163                                      Orphanet  CC BY-ND 3.0  \n",
       "19164                                      Orphanet  CC BY-ND 3.0  \n",
       "19165                                      Orphanet  CC BY-ND 3.0  \n",
       "19166                                      Orphanet  CC BY-ND 3.0  \n",
       "19167                                      Orphanet  CC BY-ND 3.0  \n",
       "19168                                      Orphanet  CC BY-ND 3.0  \n",
       "19169                                      Orphanet  CC BY-ND 3.0  \n",
       "19170                                      Orphanet  CC BY-ND 3.0  \n",
       "19171                                      Orphanet  CC BY-ND 3.0  \n",
       "19172                                      Orphanet  CC BY-ND 3.0  \n",
       "\n",
       "[19173 rows x 4 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "short_df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
