{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import sys\n",
    "import pandas \n",
    "\n",
    "# import utils.rawgit()\n",
    "rawgit = 'https://raw.githubusercontent.com/' \n",
    "# local imports\n",
    "sys.path.insert(0, './data/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Read DO Slim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>doid_id</th>\n",
       "      <th>doid_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DOID:2531</td>\n",
       "      <td>Hematologic cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DOID:1319</td>\n",
       "      <td>Brain cancer</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     doid_id           doid_name\n",
       "0  DOID:2531  Hematologic cancer\n",
       "1  DOID:1319        Brain cancer"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "commit = '72614ade9f1cc5a5317b8f6836e1e464b31d5587'\n",
    "#url = utils.rawgit('dhimmel', 'disease-ontology', commit, 'data/slim-terms.tsv')\n",
    "url = rawgit + 'dhimmel/' + 'disease-ontology/' + commit + '/data/slim-terms.tsv'\n",
    "disease_df = pandas.read_table(url)\n",
    "disease_df = disease_df.rename(columns={'doid': 'doid_id', 'name': 'doid_name'})\n",
    "disease_df = disease_df[['doid_id', 'doid_name']]\n",
    "disease_df.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Read Entrez Gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>entrez_gene_id</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1BG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   entrez_gene_id gene_symbol\n",
       "0               1        A1BG\n",
       "1               2         A2M"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#commit = '6e133f9ef8ce51a4c5387e58a6cc97564a66cec8'\n",
    "##url = utils.rawgit('dhimmel', 'entrez-gene', commit, 'data/genes-human.tsv')\n",
    "#url = rawgit + 'dhimmel/' + 'entrez-gene/' + commit + '/data/genes-human.tsv'\n",
    "gene_df = pandas.read_table('../entrez-gene/data/genes-human.tsv')\n",
    "gene_df = gene_df[gene_df.type_of_gene == 'protein-coding']\n",
    "gene_df = gene_df.rename(columns={'GeneID': 'entrez_gene_id', 'Symbol': 'gene_symbol'})\n",
    "gene_df = gene_df[['entrez_gene_id', 'gene_symbol']]\n",
    "gene_df.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Read Datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orphanet_code</th>\n",
       "      <th>orphanet_term</th>\n",
       "      <th>orphanet_gene_id</th>\n",
       "      <th>gene_orphanumber</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>dga_type</th>\n",
       "      <th>dga_status</th>\n",
       "      <th>entrez_gene_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>166024</td>\n",
       "      <td>Multiple epiphyseal dysplasia, Al-Gazali type</td>\n",
       "      <td>20160</td>\n",
       "      <td>268061</td>\n",
       "      <td>kinesin family member 7</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>KIF7</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>374654.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>93</td>\n",
       "      <td>Aspartylglucosaminuria</td>\n",
       "      <td>15470</td>\n",
       "      <td>119513</td>\n",
       "      <td>aspartylglucosaminidase</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>AGA</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>175.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   orphanet_code                                  orphanet_term  \\\n",
       "0         166024  Multiple epiphyseal dysplasia, Al-Gazali type   \n",
       "1             93                         Aspartylglucosaminuria   \n",
       "\n",
       "   orphanet_gene_id  gene_orphanumber                gene_name  \\\n",
       "0             20160            268061  kinesin family member 7   \n",
       "1             15470            119513  aspartylglucosaminidase   \n",
       "\n",
       "                   gene_type gene_symbol  \\\n",
       "0  gene with protein product        KIF7   \n",
       "1  gene with protein product         AGA   \n",
       "\n",
       "                                  dga_type dga_status  entrez_gene_id  \n",
       "0  Disease-causing germline mutation(s) in   Assessed        374654.0  \n",
       "1  Disease-causing germline mutation(s) in   Assessed           175.0  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Orphanet\n",
    "orphanet_df = pandas.read_table('../data/orphanet-disease-gene.tsv')\n",
    "orphanet_df = orphanet_df.merge(gene_df, on='gene_symbol', how='left')\n",
    "orphanet_df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'20458 genes in entrez and 6806 genes in orphanet'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'{} genes in entrez and {} genes in orphanet'.format(len(gene_df), len(orphanet_df))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gene_types: ['gene with protein product' 'Non-coding RNA' 'locus']\n",
      "dga_status: ['Assessed' 'Not yet assessed']\n"
     ]
    }
   ],
   "source": [
    "print('gene_types: {}'.format(orphanet_df.gene_type.unique()))\n",
    "print('dga_status: {}'.format(orphanet_df.dga_status.unique()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orphanet_code</th>\n",
       "      <th>orphanet_term</th>\n",
       "      <th>orphanet_gene_id</th>\n",
       "      <th>gene_orphanumber</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>dga_type</th>\n",
       "      <th>dga_status</th>\n",
       "      <th>entrez_gene_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>550</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>16471</td>\n",
       "      <td>123512</td>\n",
       "      <td>mitochondrially encoded cytochrome c oxidase I</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>550</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>16472</td>\n",
       "      <td>123516</td>\n",
       "      <td>mitochondrially encoded cytochrome c oxidase II</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-CO2</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>550</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>16473</td>\n",
       "      <td>123520</td>\n",
       "      <td>mitochondrially encoded cytochrome c oxidase III</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-CO3</td>\n",
       "      <td>Candidate gene tested in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>550</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>16478</td>\n",
       "      <td>123537</td>\n",
       "      <td>mitochondrially encoded NADH:ubiquinone oxidor...</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-ND1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>550</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>16481</td>\n",
       "      <td>123552</td>\n",
       "      <td>mitochondrially encoded NADH:ubiquinone oxidor...</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-ND4</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>550</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>16483</td>\n",
       "      <td>123562</td>\n",
       "      <td>mitochondrially encoded NADH:ubiquinone oxidor...</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-ND5</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>550</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>16484</td>\n",
       "      <td>123567</td>\n",
       "      <td>mitochondrially encoded NADH:ubiquinone oxidor...</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-ND6</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>550</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>16841</td>\n",
       "      <td>138895</td>\n",
       "      <td>mitochondrially encoded tRNA leucine 1 (UUA/G)</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TL1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>550</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>16844</td>\n",
       "      <td>138906</td>\n",
       "      <td>mitochondrially encoded tRNA glutamine</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TQ</td>\n",
       "      <td>Candidate gene tested in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>550</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>17473</td>\n",
       "      <td>160307</td>\n",
       "      <td>mitochondrially encoded tRNA tryptophan</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TW</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>550</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>17671</td>\n",
       "      <td>166766</td>\n",
       "      <td>mitochondrially encoded tRNA histidine</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TH</td>\n",
       "      <td>Candidate gene tested in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>550</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>17722</td>\n",
       "      <td>167901</td>\n",
       "      <td>mitochondrially encoded tRNA serine 1 (UCN)</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TS1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>550</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>17723</td>\n",
       "      <td>167905</td>\n",
       "      <td>mitochondrially encoded tRNA serine 2 (AGU/C)</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TS2</td>\n",
       "      <td>Candidate gene tested in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>550</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>17724</td>\n",
       "      <td>167909</td>\n",
       "      <td>mitochondrially encoded tRNA phenylalanine</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TF</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>269</td>\n",
       "      <td>Facioscapulohumeral dystrophy</td>\n",
       "      <td>15901</td>\n",
       "      <td>121239</td>\n",
       "      <td>double homeobox 4 like 1</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>DUX4L1</td>\n",
       "      <td>Candidate gene tested in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>480</td>\n",
       "      <td>Kearns-Sayre syndrome</td>\n",
       "      <td>16841</td>\n",
       "      <td>138895</td>\n",
       "      <td>mitochondrially encoded tRNA leucine 1 (UUA/G)</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TL1</td>\n",
       "      <td>Candidate gene tested in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>480</td>\n",
       "      <td>Kearns-Sayre syndrome</td>\n",
       "      <td>17396</td>\n",
       "      <td>159773</td>\n",
       "      <td>mitochondrially encoded ATP synthase 8</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-ATP8</td>\n",
       "      <td>Candidate gene tested in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>551</td>\n",
       "      <td>MERRF</td>\n",
       "      <td>16483</td>\n",
       "      <td>123562</td>\n",
       "      <td>mitochondrially encoded NADH:ubiquinone oxidor...</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-ND5</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>551</td>\n",
       "      <td>MERRF</td>\n",
       "      <td>16841</td>\n",
       "      <td>138895</td>\n",
       "      <td>mitochondrially encoded tRNA leucine 1 (UUA/G)</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TL1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>551</td>\n",
       "      <td>MERRF</td>\n",
       "      <td>16842</td>\n",
       "      <td>138900</td>\n",
       "      <td>mitochondrially encoded tRNA lysine</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TK</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>551</td>\n",
       "      <td>MERRF</td>\n",
       "      <td>16843</td>\n",
       "      <td>138902</td>\n",
       "      <td>mitochondrially encoded 12S RNA</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>Candidate gene tested in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>551</td>\n",
       "      <td>MERRF</td>\n",
       "      <td>16844</td>\n",
       "      <td>138906</td>\n",
       "      <td>mitochondrially encoded tRNA glutamine</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TQ</td>\n",
       "      <td>Candidate gene tested in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>551</td>\n",
       "      <td>MERRF</td>\n",
       "      <td>17671</td>\n",
       "      <td>166766</td>\n",
       "      <td>mitochondrially encoded tRNA histidine</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TH</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>551</td>\n",
       "      <td>MERRF</td>\n",
       "      <td>17722</td>\n",
       "      <td>167901</td>\n",
       "      <td>mitochondrially encoded tRNA serine 1 (UCN)</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TS1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>551</td>\n",
       "      <td>MERRF</td>\n",
       "      <td>17723</td>\n",
       "      <td>167905</td>\n",
       "      <td>mitochondrially encoded tRNA serine 2 (AGU/C)</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TS2</td>\n",
       "      <td>Candidate gene tested in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>551</td>\n",
       "      <td>MERRF</td>\n",
       "      <td>17724</td>\n",
       "      <td>167909</td>\n",
       "      <td>mitochondrially encoded tRNA phenylalanine</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TF</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>551</td>\n",
       "      <td>MERRF</td>\n",
       "      <td>18493</td>\n",
       "      <td>205934</td>\n",
       "      <td>mitochondrially encoded tRNA proline</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TP</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>803</td>\n",
       "      <td>Amyotrophic lateral sclerosis</td>\n",
       "      <td>20676</td>\n",
       "      <td>286546</td>\n",
       "      <td>chromosome 9 open reading frame 72</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>C9ORF72</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>567</td>\n",
       "      <td>22q11.2 deletion syndrome</td>\n",
       "      <td>17262</td>\n",
       "      <td>159084</td>\n",
       "      <td>ubiquitin fusion degradation 1 like (yeast)</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>UFD1L</td>\n",
       "      <td>Role in the phenotype of</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197</th>\n",
       "      <td>104</td>\n",
       "      <td>Leber hereditary optic neuropathy</td>\n",
       "      <td>16470</td>\n",
       "      <td>123508</td>\n",
       "      <td>mitochondrially encoded ATP synthase 6</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-ATP6</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5657</th>\n",
       "      <td>90641</td>\n",
       "      <td>Mitochondrial non-syndromic sensorineural deaf...</td>\n",
       "      <td>17671</td>\n",
       "      <td>166766</td>\n",
       "      <td>mitochondrially encoded tRNA histidine</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TH</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5658</th>\n",
       "      <td>90641</td>\n",
       "      <td>Mitochondrial non-syndromic sensorineural deaf...</td>\n",
       "      <td>17722</td>\n",
       "      <td>167901</td>\n",
       "      <td>mitochondrially encoded tRNA serine 1 (UCN)</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MT-TS1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5979</th>\n",
       "      <td>96191</td>\n",
       "      <td>Paternal uniparental disomy of chromosome 6</td>\n",
       "      <td>17383</td>\n",
       "      <td>159723</td>\n",
       "      <td>hydatidiform mole associated and imprinted (no...</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>HYMAI</td>\n",
       "      <td>Role in the phenotype of</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5981</th>\n",
       "      <td>96184</td>\n",
       "      <td>Maternal uniparental disomy of chromosome 14</td>\n",
       "      <td>17275</td>\n",
       "      <td>159158</td>\n",
       "      <td>maternally expressed 3 (non-protein coding)</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MEG3</td>\n",
       "      <td>Role in the phenotype of</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5987</th>\n",
       "      <td>96334</td>\n",
       "      <td>Paternal uniparental disomy of chromosome 14</td>\n",
       "      <td>17275</td>\n",
       "      <td>159158</td>\n",
       "      <td>maternally expressed 3 (non-protein coding)</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>MEG3</td>\n",
       "      <td>Role in the phenotype of</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6208</th>\n",
       "      <td>99718</td>\n",
       "      <td>Leber plus disease</td>\n",
       "      <td>16480</td>\n",
       "      <td>123547</td>\n",
       "      <td>mitochondrially encoded NADH:ubiquinone oxidor...</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-ND3</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6209</th>\n",
       "      <td>99718</td>\n",
       "      <td>Leber plus disease</td>\n",
       "      <td>16481</td>\n",
       "      <td>123552</td>\n",
       "      <td>mitochondrially encoded NADH:ubiquinone oxidor...</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-ND4</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6210</th>\n",
       "      <td>99718</td>\n",
       "      <td>Leber plus disease</td>\n",
       "      <td>16484</td>\n",
       "      <td>123567</td>\n",
       "      <td>mitochondrially encoded NADH:ubiquinone oxidor...</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-ND6</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6234</th>\n",
       "      <td>98807</td>\n",
       "      <td>Primary dystonia, DYT13 type</td>\n",
       "      <td>17437</td>\n",
       "      <td>160001</td>\n",
       "      <td>dystonia 13, torsion</td>\n",
       "      <td>locus</td>\n",
       "      <td>DYT13</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6266</th>\n",
       "      <td>98869</td>\n",
       "      <td>Congenital dyserythropoietic anemia type I</td>\n",
       "      <td>22273</td>\n",
       "      <td>362774</td>\n",
       "      <td>chromosome 15 open reading frame 41</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>C15ORF41</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6287</th>\n",
       "      <td>98890</td>\n",
       "      <td>Early-onset X-linked optic atrophy</td>\n",
       "      <td>21418</td>\n",
       "      <td>310868</td>\n",
       "      <td>optic atrophy 2 (obscure)</td>\n",
       "      <td>locus</td>\n",
       "      <td>OPA2</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6492</th>\n",
       "      <td>98760</td>\n",
       "      <td>Spinocerebellar ataxia type 8</td>\n",
       "      <td>17197</td>\n",
       "      <td>158384</td>\n",
       "      <td>ATXN8 opposite strand (non-protein coding)</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>ATXN8OS</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6525</th>\n",
       "      <td>101111</td>\n",
       "      <td>Spinocerebellar ataxia type 25</td>\n",
       "      <td>18725</td>\n",
       "      <td>212612</td>\n",
       "      <td>spinocerebellar ataxia 25</td>\n",
       "      <td>locus</td>\n",
       "      <td>SCA25</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6528</th>\n",
       "      <td>101110</td>\n",
       "      <td>Spinocerebellar ataxia type 20</td>\n",
       "      <td>18724</td>\n",
       "      <td>212610</td>\n",
       "      <td>spinocerebellar ataxia 20</td>\n",
       "      <td>locus</td>\n",
       "      <td>SCA20</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6563</th>\n",
       "      <td>101009</td>\n",
       "      <td>Autosomal dominant spastic paraplegia type 29</td>\n",
       "      <td>18127</td>\n",
       "      <td>180232</td>\n",
       "      <td>spastic paraplegia 29 (autosomal dominant)</td>\n",
       "      <td>locus</td>\n",
       "      <td>SPG29</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6588</th>\n",
       "      <td>100999</td>\n",
       "      <td>Autosomal dominant spastic paraplegia type 19</td>\n",
       "      <td>18121</td>\n",
       "      <td>180213</td>\n",
       "      <td>spastic paraplegia 19 (autosomal dominant)</td>\n",
       "      <td>locus</td>\n",
       "      <td>SPG19</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6590</th>\n",
       "      <td>100997</td>\n",
       "      <td>X-linked spastic paraplegia type 16</td>\n",
       "      <td>18153</td>\n",
       "      <td>180322</td>\n",
       "      <td>spastic paraplegia 16 (complicated, X-linked r...</td>\n",
       "      <td>locus</td>\n",
       "      <td>SPG16</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6592</th>\n",
       "      <td>100995</td>\n",
       "      <td>Autosomal recessive spastic paraplegia type 14</td>\n",
       "      <td>18144</td>\n",
       "      <td>180294</td>\n",
       "      <td>spastic paraplegia 14 (autosomal recessive)</td>\n",
       "      <td>locus</td>\n",
       "      <td>SPG14</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6596</th>\n",
       "      <td>101007</td>\n",
       "      <td>Autosomal recessive spastic paraplegia type 27</td>\n",
       "      <td>18150</td>\n",
       "      <td>180316</td>\n",
       "      <td>spastic paraplegia 27 (autosomal recessive)</td>\n",
       "      <td>locus</td>\n",
       "      <td>SPG27</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6598</th>\n",
       "      <td>101005</td>\n",
       "      <td>Autosomal recessive spastic paraplegia type 25</td>\n",
       "      <td>18147</td>\n",
       "      <td>180307</td>\n",
       "      <td>spastic paraplegia 25 (autosomal recessive, wi...</td>\n",
       "      <td>locus</td>\n",
       "      <td>SPG25</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6600</th>\n",
       "      <td>101003</td>\n",
       "      <td>Autosomal recessive spastic paraplegia type 23</td>\n",
       "      <td>18145</td>\n",
       "      <td>180301</td>\n",
       "      <td>spastic paraplegia 23 (autosomal recessive)</td>\n",
       "      <td>locus</td>\n",
       "      <td>SPG23</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6601</th>\n",
       "      <td>101004</td>\n",
       "      <td>Autosomal recessive spastic paraplegia type 24</td>\n",
       "      <td>18139</td>\n",
       "      <td>180270</td>\n",
       "      <td>spastic paraplegia 24 (autosomal recessive)</td>\n",
       "      <td>locus</td>\n",
       "      <td>SPG24</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6606</th>\n",
       "      <td>100973</td>\n",
       "      <td>FRAXE intellectual disability</td>\n",
       "      <td>17364</td>\n",
       "      <td>159590</td>\n",
       "      <td>fragile X mental retardation associated 3</td>\n",
       "      <td>locus</td>\n",
       "      <td>FMR3</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6618</th>\n",
       "      <td>100069</td>\n",
       "      <td>Semantic dementia</td>\n",
       "      <td>20676</td>\n",
       "      <td>286546</td>\n",
       "      <td>chromosome 9 open reading frame 72</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>C9ORF72</td>\n",
       "      <td>Major susceptibility factor in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6625</th>\n",
       "      <td>100070</td>\n",
       "      <td>Progressive non-fluent aphasia</td>\n",
       "      <td>20676</td>\n",
       "      <td>286546</td>\n",
       "      <td>chromosome 9 open reading frame 72</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>C9ORF72</td>\n",
       "      <td>Major susceptibility factor in</td>\n",
       "      <td>Not yet assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6638</th>\n",
       "      <td>100033</td>\n",
       "      <td>Hypomaturation amelogenesis imperfecta</td>\n",
       "      <td>21411</td>\n",
       "      <td>310402</td>\n",
       "      <td>chromosome 4 open reading frame 26</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>C4ORF26</td>\n",
       "      <td>Disease-causing germline mutation(s) (loss of ...</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6702</th>\n",
       "      <td>99886</td>\n",
       "      <td>Transient neonatal diabetes mellitus</td>\n",
       "      <td>17383</td>\n",
       "      <td>159723</td>\n",
       "      <td>hydatidiform mole associated and imprinted (no...</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>HYMAI</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6749</th>\n",
       "      <td>99845</td>\n",
       "      <td>Genetic recurrent myoglobinuria</td>\n",
       "      <td>16471</td>\n",
       "      <td>123512</td>\n",
       "      <td>mitochondrially encoded cytochrome c oxidase I</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6750</th>\n",
       "      <td>99845</td>\n",
       "      <td>Genetic recurrent myoglobinuria</td>\n",
       "      <td>16473</td>\n",
       "      <td>123520</td>\n",
       "      <td>mitochondrially encoded cytochrome c oxidase III</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-CO3</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6765</th>\n",
       "      <td>99852</td>\n",
       "      <td>Ravine syndrome</td>\n",
       "      <td>22738</td>\n",
       "      <td>398677</td>\n",
       "      <td>solute carrier family 7, member 2-intron1 (lon...</td>\n",
       "      <td>Non-coding RNA</td>\n",
       "      <td>SLC7A2-IT1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>197 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      orphanet_code                                      orphanet_term  \\\n",
       "41              550                                              MELAS   \n",
       "42              550                                              MELAS   \n",
       "43              550                                              MELAS   \n",
       "44              550                                              MELAS   \n",
       "45              550                                              MELAS   \n",
       "46              550                                              MELAS   \n",
       "47              550                                              MELAS   \n",
       "48              550                                              MELAS   \n",
       "49              550                                              MELAS   \n",
       "50              550                                              MELAS   \n",
       "51              550                                              MELAS   \n",
       "52              550                                              MELAS   \n",
       "53              550                                              MELAS   \n",
       "54              550                                              MELAS   \n",
       "55              269                      Facioscapulohumeral dystrophy   \n",
       "60              480                              Kearns-Sayre syndrome   \n",
       "61              480                              Kearns-Sayre syndrome   \n",
       "63              551                                              MERRF   \n",
       "64              551                                              MERRF   \n",
       "65              551                                              MERRF   \n",
       "66              551                                              MERRF   \n",
       "67              551                                              MERRF   \n",
       "68              551                                              MERRF   \n",
       "69              551                                              MERRF   \n",
       "70              551                                              MERRF   \n",
       "71              551                                              MERRF   \n",
       "72              551                                              MERRF   \n",
       "126             803                      Amyotrophic lateral sclerosis   \n",
       "145             567                          22q11.2 deletion syndrome   \n",
       "197             104                  Leber hereditary optic neuropathy   \n",
       "...             ...                                                ...   \n",
       "5657          90641  Mitochondrial non-syndromic sensorineural deaf...   \n",
       "5658          90641  Mitochondrial non-syndromic sensorineural deaf...   \n",
       "5979          96191        Paternal uniparental disomy of chromosome 6   \n",
       "5981          96184       Maternal uniparental disomy of chromosome 14   \n",
       "5987          96334       Paternal uniparental disomy of chromosome 14   \n",
       "6208          99718                                 Leber plus disease   \n",
       "6209          99718                                 Leber plus disease   \n",
       "6210          99718                                 Leber plus disease   \n",
       "6234          98807                       Primary dystonia, DYT13 type   \n",
       "6266          98869         Congenital dyserythropoietic anemia type I   \n",
       "6287          98890                 Early-onset X-linked optic atrophy   \n",
       "6492          98760                      Spinocerebellar ataxia type 8   \n",
       "6525         101111                     Spinocerebellar ataxia type 25   \n",
       "6528         101110                     Spinocerebellar ataxia type 20   \n",
       "6563         101009      Autosomal dominant spastic paraplegia type 29   \n",
       "6588         100999      Autosomal dominant spastic paraplegia type 19   \n",
       "6590         100997                X-linked spastic paraplegia type 16   \n",
       "6592         100995     Autosomal recessive spastic paraplegia type 14   \n",
       "6596         101007     Autosomal recessive spastic paraplegia type 27   \n",
       "6598         101005     Autosomal recessive spastic paraplegia type 25   \n",
       "6600         101003     Autosomal recessive spastic paraplegia type 23   \n",
       "6601         101004     Autosomal recessive spastic paraplegia type 24   \n",
       "6606         100973                      FRAXE intellectual disability   \n",
       "6618         100069                                  Semantic dementia   \n",
       "6625         100070                     Progressive non-fluent aphasia   \n",
       "6638         100033             Hypomaturation amelogenesis imperfecta   \n",
       "6702          99886               Transient neonatal diabetes mellitus   \n",
       "6749          99845                    Genetic recurrent myoglobinuria   \n",
       "6750          99845                    Genetic recurrent myoglobinuria   \n",
       "6765          99852                                    Ravine syndrome   \n",
       "\n",
       "      orphanet_gene_id  gene_orphanumber  \\\n",
       "41               16471            123512   \n",
       "42               16472            123516   \n",
       "43               16473            123520   \n",
       "44               16478            123537   \n",
       "45               16481            123552   \n",
       "46               16483            123562   \n",
       "47               16484            123567   \n",
       "48               16841            138895   \n",
       "49               16844            138906   \n",
       "50               17473            160307   \n",
       "51               17671            166766   \n",
       "52               17722            167901   \n",
       "53               17723            167905   \n",
       "54               17724            167909   \n",
       "55               15901            121239   \n",
       "60               16841            138895   \n",
       "61               17396            159773   \n",
       "63               16483            123562   \n",
       "64               16841            138895   \n",
       "65               16842            138900   \n",
       "66               16843            138902   \n",
       "67               16844            138906   \n",
       "68               17671            166766   \n",
       "69               17722            167901   \n",
       "70               17723            167905   \n",
       "71               17724            167909   \n",
       "72               18493            205934   \n",
       "126              20676            286546   \n",
       "145              17262            159084   \n",
       "197              16470            123508   \n",
       "...                ...               ...   \n",
       "5657             17671            166766   \n",
       "5658             17722            167901   \n",
       "5979             17383            159723   \n",
       "5981             17275            159158   \n",
       "5987             17275            159158   \n",
       "6208             16480            123547   \n",
       "6209             16481            123552   \n",
       "6210             16484            123567   \n",
       "6234             17437            160001   \n",
       "6266             22273            362774   \n",
       "6287             21418            310868   \n",
       "6492             17197            158384   \n",
       "6525             18725            212612   \n",
       "6528             18724            212610   \n",
       "6563             18127            180232   \n",
       "6588             18121            180213   \n",
       "6590             18153            180322   \n",
       "6592             18144            180294   \n",
       "6596             18150            180316   \n",
       "6598             18147            180307   \n",
       "6600             18145            180301   \n",
       "6601             18139            180270   \n",
       "6606             17364            159590   \n",
       "6618             20676            286546   \n",
       "6625             20676            286546   \n",
       "6638             21411            310402   \n",
       "6702             17383            159723   \n",
       "6749             16471            123512   \n",
       "6750             16473            123520   \n",
       "6765             22738            398677   \n",
       "\n",
       "                                              gene_name  \\\n",
       "41       mitochondrially encoded cytochrome c oxidase I   \n",
       "42      mitochondrially encoded cytochrome c oxidase II   \n",
       "43     mitochondrially encoded cytochrome c oxidase III   \n",
       "44    mitochondrially encoded NADH:ubiquinone oxidor...   \n",
       "45    mitochondrially encoded NADH:ubiquinone oxidor...   \n",
       "46    mitochondrially encoded NADH:ubiquinone oxidor...   \n",
       "47    mitochondrially encoded NADH:ubiquinone oxidor...   \n",
       "48       mitochondrially encoded tRNA leucine 1 (UUA/G)   \n",
       "49               mitochondrially encoded tRNA glutamine   \n",
       "50              mitochondrially encoded tRNA tryptophan   \n",
       "51               mitochondrially encoded tRNA histidine   \n",
       "52          mitochondrially encoded tRNA serine 1 (UCN)   \n",
       "53        mitochondrially encoded tRNA serine 2 (AGU/C)   \n",
       "54           mitochondrially encoded tRNA phenylalanine   \n",
       "55                             double homeobox 4 like 1   \n",
       "60       mitochondrially encoded tRNA leucine 1 (UUA/G)   \n",
       "61               mitochondrially encoded ATP synthase 8   \n",
       "63    mitochondrially encoded NADH:ubiquinone oxidor...   \n",
       "64       mitochondrially encoded tRNA leucine 1 (UUA/G)   \n",
       "65                  mitochondrially encoded tRNA lysine   \n",
       "66                      mitochondrially encoded 12S RNA   \n",
       "67               mitochondrially encoded tRNA glutamine   \n",
       "68               mitochondrially encoded tRNA histidine   \n",
       "69          mitochondrially encoded tRNA serine 1 (UCN)   \n",
       "70        mitochondrially encoded tRNA serine 2 (AGU/C)   \n",
       "71           mitochondrially encoded tRNA phenylalanine   \n",
       "72                 mitochondrially encoded tRNA proline   \n",
       "126                  chromosome 9 open reading frame 72   \n",
       "145         ubiquitin fusion degradation 1 like (yeast)   \n",
       "197              mitochondrially encoded ATP synthase 6   \n",
       "...                                                 ...   \n",
       "5657             mitochondrially encoded tRNA histidine   \n",
       "5658        mitochondrially encoded tRNA serine 1 (UCN)   \n",
       "5979  hydatidiform mole associated and imprinted (no...   \n",
       "5981        maternally expressed 3 (non-protein coding)   \n",
       "5987        maternally expressed 3 (non-protein coding)   \n",
       "6208  mitochondrially encoded NADH:ubiquinone oxidor...   \n",
       "6209  mitochondrially encoded NADH:ubiquinone oxidor...   \n",
       "6210  mitochondrially encoded NADH:ubiquinone oxidor...   \n",
       "6234                               dystonia 13, torsion   \n",
       "6266                chromosome 15 open reading frame 41   \n",
       "6287                          optic atrophy 2 (obscure)   \n",
       "6492         ATXN8 opposite strand (non-protein coding)   \n",
       "6525                          spinocerebellar ataxia 25   \n",
       "6528                          spinocerebellar ataxia 20   \n",
       "6563         spastic paraplegia 29 (autosomal dominant)   \n",
       "6588         spastic paraplegia 19 (autosomal dominant)   \n",
       "6590  spastic paraplegia 16 (complicated, X-linked r...   \n",
       "6592        spastic paraplegia 14 (autosomal recessive)   \n",
       "6596        spastic paraplegia 27 (autosomal recessive)   \n",
       "6598  spastic paraplegia 25 (autosomal recessive, wi...   \n",
       "6600        spastic paraplegia 23 (autosomal recessive)   \n",
       "6601        spastic paraplegia 24 (autosomal recessive)   \n",
       "6606          fragile X mental retardation associated 3   \n",
       "6618                 chromosome 9 open reading frame 72   \n",
       "6625                 chromosome 9 open reading frame 72   \n",
       "6638                 chromosome 4 open reading frame 26   \n",
       "6702  hydatidiform mole associated and imprinted (no...   \n",
       "6749     mitochondrially encoded cytochrome c oxidase I   \n",
       "6750   mitochondrially encoded cytochrome c oxidase III   \n",
       "6765  solute carrier family 7, member 2-intron1 (lon...   \n",
       "\n",
       "                      gene_type gene_symbol  \\\n",
       "41    gene with protein product      MT-CO1   \n",
       "42    gene with protein product      MT-CO2   \n",
       "43    gene with protein product      MT-CO3   \n",
       "44    gene with protein product      MT-ND1   \n",
       "45    gene with protein product      MT-ND4   \n",
       "46    gene with protein product      MT-ND5   \n",
       "47    gene with protein product      MT-ND6   \n",
       "48               Non-coding RNA      MT-TL1   \n",
       "49               Non-coding RNA       MT-TQ   \n",
       "50               Non-coding RNA       MT-TW   \n",
       "51               Non-coding RNA       MT-TH   \n",
       "52               Non-coding RNA      MT-TS1   \n",
       "53               Non-coding RNA      MT-TS2   \n",
       "54               Non-coding RNA       MT-TF   \n",
       "55    gene with protein product      DUX4L1   \n",
       "60               Non-coding RNA      MT-TL1   \n",
       "61    gene with protein product     MT-ATP8   \n",
       "63    gene with protein product      MT-ND5   \n",
       "64               Non-coding RNA      MT-TL1   \n",
       "65               Non-coding RNA       MT-TK   \n",
       "66               Non-coding RNA     MT-RNR1   \n",
       "67               Non-coding RNA       MT-TQ   \n",
       "68               Non-coding RNA       MT-TH   \n",
       "69               Non-coding RNA      MT-TS1   \n",
       "70               Non-coding RNA      MT-TS2   \n",
       "71               Non-coding RNA       MT-TF   \n",
       "72               Non-coding RNA       MT-TP   \n",
       "126   gene with protein product     C9ORF72   \n",
       "145   gene with protein product       UFD1L   \n",
       "197   gene with protein product     MT-ATP6   \n",
       "...                         ...         ...   \n",
       "5657             Non-coding RNA       MT-TH   \n",
       "5658             Non-coding RNA      MT-TS1   \n",
       "5979             Non-coding RNA       HYMAI   \n",
       "5981             Non-coding RNA        MEG3   \n",
       "5987             Non-coding RNA        MEG3   \n",
       "6208  gene with protein product      MT-ND3   \n",
       "6209  gene with protein product      MT-ND4   \n",
       "6210  gene with protein product      MT-ND6   \n",
       "6234                      locus       DYT13   \n",
       "6266  gene with protein product    C15ORF41   \n",
       "6287                      locus        OPA2   \n",
       "6492             Non-coding RNA     ATXN8OS   \n",
       "6525                      locus       SCA25   \n",
       "6528                      locus       SCA20   \n",
       "6563                      locus       SPG29   \n",
       "6588                      locus       SPG19   \n",
       "6590                      locus       SPG16   \n",
       "6592                      locus       SPG14   \n",
       "6596                      locus       SPG27   \n",
       "6598                      locus       SPG25   \n",
       "6600                      locus       SPG23   \n",
       "6601                      locus       SPG24   \n",
       "6606                      locus        FMR3   \n",
       "6618  gene with protein product     C9ORF72   \n",
       "6625  gene with protein product     C9ORF72   \n",
       "6638  gene with protein product     C4ORF26   \n",
       "6702             Non-coding RNA       HYMAI   \n",
       "6749  gene with protein product      MT-CO1   \n",
       "6750  gene with protein product      MT-CO3   \n",
       "6765             Non-coding RNA  SLC7A2-IT1   \n",
       "\n",
       "                                               dga_type        dga_status  \\\n",
       "41              Disease-causing germline mutation(s) in          Assessed   \n",
       "42              Disease-causing germline mutation(s) in          Assessed   \n",
       "43                             Candidate gene tested in  Not yet assessed   \n",
       "44              Disease-causing germline mutation(s) in          Assessed   \n",
       "45              Disease-causing germline mutation(s) in          Assessed   \n",
       "46              Disease-causing germline mutation(s) in          Assessed   \n",
       "47              Disease-causing germline mutation(s) in          Assessed   \n",
       "48              Disease-causing germline mutation(s) in          Assessed   \n",
       "49                             Candidate gene tested in  Not yet assessed   \n",
       "50              Disease-causing germline mutation(s) in          Assessed   \n",
       "51                             Candidate gene tested in  Not yet assessed   \n",
       "52              Disease-causing germline mutation(s) in          Assessed   \n",
       "53                             Candidate gene tested in  Not yet assessed   \n",
       "54              Disease-causing germline mutation(s) in          Assessed   \n",
       "55                             Candidate gene tested in  Not yet assessed   \n",
       "60                             Candidate gene tested in  Not yet assessed   \n",
       "61                             Candidate gene tested in  Not yet assessed   \n",
       "63              Disease-causing germline mutation(s) in          Assessed   \n",
       "64              Disease-causing germline mutation(s) in          Assessed   \n",
       "65              Disease-causing germline mutation(s) in          Assessed   \n",
       "66                             Candidate gene tested in  Not yet assessed   \n",
       "67                             Candidate gene tested in  Not yet assessed   \n",
       "68              Disease-causing germline mutation(s) in          Assessed   \n",
       "69              Disease-causing germline mutation(s) in          Assessed   \n",
       "70                             Candidate gene tested in  Not yet assessed   \n",
       "71              Disease-causing germline mutation(s) in          Assessed   \n",
       "72              Disease-causing germline mutation(s) in          Assessed   \n",
       "126             Disease-causing germline mutation(s) in          Assessed   \n",
       "145                            Role in the phenotype of          Assessed   \n",
       "197             Disease-causing germline mutation(s) in          Assessed   \n",
       "...                                                 ...               ...   \n",
       "5657            Disease-causing germline mutation(s) in          Assessed   \n",
       "5658            Disease-causing germline mutation(s) in          Assessed   \n",
       "5979                           Role in the phenotype of  Not yet assessed   \n",
       "5981                           Role in the phenotype of          Assessed   \n",
       "5987                           Role in the phenotype of          Assessed   \n",
       "6208            Disease-causing germline mutation(s) in  Not yet assessed   \n",
       "6209            Disease-causing germline mutation(s) in  Not yet assessed   \n",
       "6210            Disease-causing germline mutation(s) in          Assessed   \n",
       "6234            Disease-causing germline mutation(s) in          Assessed   \n",
       "6266            Disease-causing germline mutation(s) in          Assessed   \n",
       "6287            Disease-causing germline mutation(s) in          Assessed   \n",
       "6492            Disease-causing germline mutation(s) in          Assessed   \n",
       "6525            Disease-causing germline mutation(s) in          Assessed   \n",
       "6528            Disease-causing germline mutation(s) in          Assessed   \n",
       "6563            Disease-causing germline mutation(s) in  Not yet assessed   \n",
       "6588            Disease-causing germline mutation(s) in          Assessed   \n",
       "6590            Disease-causing germline mutation(s) in          Assessed   \n",
       "6592            Disease-causing germline mutation(s) in          Assessed   \n",
       "6596            Disease-causing germline mutation(s) in          Assessed   \n",
       "6598            Disease-causing germline mutation(s) in          Assessed   \n",
       "6600            Disease-causing germline mutation(s) in          Assessed   \n",
       "6601            Disease-causing germline mutation(s) in          Assessed   \n",
       "6606            Disease-causing germline mutation(s) in          Assessed   \n",
       "6618                     Major susceptibility factor in  Not yet assessed   \n",
       "6625                     Major susceptibility factor in  Not yet assessed   \n",
       "6638  Disease-causing germline mutation(s) (loss of ...          Assessed   \n",
       "6702            Disease-causing germline mutation(s) in          Assessed   \n",
       "6749            Disease-causing germline mutation(s) in          Assessed   \n",
       "6750            Disease-causing germline mutation(s) in          Assessed   \n",
       "6765            Disease-causing germline mutation(s) in          Assessed   \n",
       "\n",
       "     entrez_gene_id  \n",
       "41          missing  \n",
       "42          missing  \n",
       "43          missing  \n",
       "44          missing  \n",
       "45          missing  \n",
       "46          missing  \n",
       "47          missing  \n",
       "48          missing  \n",
       "49          missing  \n",
       "50          missing  \n",
       "51          missing  \n",
       "52          missing  \n",
       "53          missing  \n",
       "54          missing  \n",
       "55          missing  \n",
       "60          missing  \n",
       "61          missing  \n",
       "63          missing  \n",
       "64          missing  \n",
       "65          missing  \n",
       "66          missing  \n",
       "67          missing  \n",
       "68          missing  \n",
       "69          missing  \n",
       "70          missing  \n",
       "71          missing  \n",
       "72          missing  \n",
       "126         missing  \n",
       "145         missing  \n",
       "197         missing  \n",
       "...             ...  \n",
       "5657        missing  \n",
       "5658        missing  \n",
       "5979        missing  \n",
       "5981        missing  \n",
       "5987        missing  \n",
       "6208        missing  \n",
       "6209        missing  \n",
       "6210        missing  \n",
       "6234        missing  \n",
       "6266        missing  \n",
       "6287        missing  \n",
       "6492        missing  \n",
       "6525        missing  \n",
       "6528        missing  \n",
       "6563        missing  \n",
       "6588        missing  \n",
       "6590        missing  \n",
       "6592        missing  \n",
       "6596        missing  \n",
       "6598        missing  \n",
       "6600        missing  \n",
       "6601        missing  \n",
       "6606        missing  \n",
       "6618        missing  \n",
       "6625        missing  \n",
       "6638        missing  \n",
       "6702        missing  \n",
       "6749        missing  \n",
       "6750        missing  \n",
       "6765        missing  \n",
       "\n",
       "[197 rows x 10 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check if there are missing values\n",
    "orpha_na = orphanet_df.fillna('missing')\n",
    "orpha_na.query('entrez_gene_id == \"missing\"')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'NA GeneID: 197'"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'NA GeneID: {}'.format(len(orpha_na.query('entrez_gene_id == \"missing\"')))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'NA GeneID for protein codeing: 88'"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'NA GeneID for protein codeing: {}'.format(len(orpha_na.query('entrez_gene_id == \"missing\" and gene_type == \"gene with protein product\"')))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>orphanet_code</th>\n",
       "      <th>orphanet_term</th>\n",
       "      <th>orphanet_gene_id</th>\n",
       "      <th>gene_orphanumber</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>dga_type</th>\n",
       "      <th>dga_status</th>\n",
       "      <th>entrez_gene_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>550</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>16471</td>\n",
       "      <td>123512</td>\n",
       "      <td>mitochondrially encoded cytochrome c oxidase I</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>550</td>\n",
       "      <td>MELAS</td>\n",
       "      <td>16472</td>\n",
       "      <td>123516</td>\n",
       "      <td>mitochondrially encoded cytochrome c oxidase II</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>MT-CO2</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>missing</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    orphanet_code orphanet_term  orphanet_gene_id  gene_orphanumber  \\\n",
       "41            550         MELAS             16471            123512   \n",
       "42            550         MELAS             16472            123516   \n",
       "\n",
       "                                          gene_name  \\\n",
       "41   mitochondrially encoded cytochrome c oxidase I   \n",
       "42  mitochondrially encoded cytochrome c oxidase II   \n",
       "\n",
       "                    gene_type gene_symbol  \\\n",
       "41  gene with protein product      MT-CO1   \n",
       "42  gene with protein product      MT-CO2   \n",
       "\n",
       "                                   dga_type dga_status entrez_gene_id  \n",
       "41  Disease-causing germline mutation(s) in   Assessed        missing  \n",
       "42  Disease-causing germline mutation(s) in   Assessed        missing  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "orpha_na.query('entrez_gene_id == \"missing\" and gene_type == \"gene with protein product\"').head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>entrez_gene_id</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [entrez_gene_id, gene_symbol]\n",
       "Index: []"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_df.query('gene_symbol == \"MT-CO1\"')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>entrez_gene_id</th>\n",
       "      <th>gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3622</th>\n",
       "      <td>4512</td>\n",
       "      <td>COX1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      entrez_gene_id gene_symbol\n",
       "3622            4512        COX1"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_df.query('entrez_gene_id == 4512')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# the problem is that in the web site the officia symbol for entrez:4512 is 'MT-CO1',\n",
    "# while in the downloaded file is 'COX1' +> try the mapping using biothings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "genes pre-removal: 6806\n",
      "genes post-removal: 6609\n"
     ]
    }
   ],
   "source": [
    "# Remove genes with missing entrez_gene_id data\n",
    "orpha_df = orpha_na.query('entrez_gene_id != \"missing\"')\n",
    "print('genes pre-removal: {}'.format(len(orpha_na)))\n",
    "print('genes post-removal: {}'.format(len(orpha_df)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>doid_id</th>\n",
       "      <th>doid_name</th>\n",
       "      <th>orphanet_gene_id</th>\n",
       "      <th>gene_orphanumber</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>dga_type</th>\n",
       "      <th>dga_status</th>\n",
       "      <th>entrez_gene_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ORPHA:166024</td>\n",
       "      <td>Multiple epiphyseal dysplasia, Al-Gazali type</td>\n",
       "      <td>20160</td>\n",
       "      <td>268061</td>\n",
       "      <td>kinesin family member 7</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>KIF7</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>374654</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ORPHA:93</td>\n",
       "      <td>Aspartylglucosaminuria</td>\n",
       "      <td>15470</td>\n",
       "      <td>119513</td>\n",
       "      <td>aspartylglucosaminidase</td>\n",
       "      <td>gene with protein product</td>\n",
       "      <td>AGA</td>\n",
       "      <td>Disease-causing germline mutation(s) in</td>\n",
       "      <td>Assessed</td>\n",
       "      <td>175</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        doid_id                                      doid_name  \\\n",
       "0  ORPHA:166024  Multiple epiphyseal dysplasia, Al-Gazali type   \n",
       "1      ORPHA:93                         Aspartylglucosaminuria   \n",
       "\n",
       "   orphanet_gene_id  gene_orphanumber                gene_name  \\\n",
       "0             20160            268061  kinesin family member 7   \n",
       "1             15470            119513  aspartylglucosaminidase   \n",
       "\n",
       "                   gene_type gene_symbol  \\\n",
       "0  gene with protein product        KIF7   \n",
       "1  gene with protein product         AGA   \n",
       "\n",
       "                                  dga_type dga_status entrez_gene_id  \n",
       "0  Disease-causing germline mutation(s) in   Assessed         374654  \n",
       "1  Disease-causing germline mutation(s) in   Assessed            175  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# reformat\n",
    "orpha_df = orpha_df.rename(columns={'orphanet_code': 'doid_id', 'orphanet_term': 'doid_name'})\n",
    "orpha_df['doid_id'] = 'ORPHA:' + orpha_df['doid_id'].astype(str)\n",
    "orpha_df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>doid_id</th>\n",
       "      <th>doid_name</th>\n",
       "      <th>entrez_gene_id</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>score_text</th>\n",
       "      <th>score_knowledge</th>\n",
       "      <th>score_cosmic</th>\n",
       "      <th>score_distild</th>\n",
       "      <th>score_integrated_no_distild</th>\n",
       "      <th>score_integrated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DOID:13223</td>\n",
       "      <td>uterine fibroid</td>\n",
       "      <td>60</td>\n",
       "      <td>ACTB</td>\n",
       "      <td>0.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DOID:13223</td>\n",
       "      <td>uterine fibroid</td>\n",
       "      <td>71</td>\n",
       "      <td>ACTG1</td>\n",
       "      <td>0.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8</td>\n",
       "      <td>0.8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      doid_id        doid_name  entrez_gene_id gene_symbol  score_text  \\\n",
       "0  DOID:13223  uterine fibroid              60        ACTB         0.8   \n",
       "1  DOID:13223  uterine fibroid              71       ACTG1         0.8   \n",
       "\n",
       "   score_knowledge  score_cosmic  score_distild  score_integrated_no_distild  \\\n",
       "0              NaN           NaN            NaN                          0.8   \n",
       "1              NaN           NaN            NaN                          0.8   \n",
       "\n",
       "   score_integrated  \n",
       "0               0.8  \n",
       "1               0.8  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# DISEASES\n",
    "commit = 'e0089ef89a56348d7d4e0684a9c51c5747b16237'\n",
    "#url = utils.rawgit('dhimmel', 'diseases', commit, 'data/merged-slim.tsv')\n",
    "url = rawgit + 'dhimmel/' + 'diseases/' + commit + '/data/merged-slim.tsv'\n",
    "diseases_df = pandas.read_table(url)\n",
    "diseases_df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>doid_id</th>\n",
       "      <th>doid_name</th>\n",
       "      <th>entrez_gene_id</th>\n",
       "      <th>Symbol</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DOID:0001816</td>\n",
       "      <td>angiosarcoma</td>\n",
       "      <td>302</td>\n",
       "      <td>ANXA2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DOID:0001816</td>\n",
       "      <td>angiosarcoma</td>\n",
       "      <td>595</td>\n",
       "      <td>CCND1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        doid_id     doid_name  entrez_gene_id Symbol  count\n",
       "0  DOID:0001816  angiosarcoma             302  ANXA2      1\n",
       "1  DOID:0001816  angiosarcoma             595  CCND1      1"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# DOAF\n",
    "commit = 'bbe1c326aa385416e36d02b144e89e2b99e700b6'\n",
    "#url = utils.rawgit('dhimmel', 'doaf', commit, 'data/doaf.tsv')\n",
    "url = rawgit + 'dhimmel/' + 'doaf/' + commit + '/data/doaf.tsv'\n",
    "doaf_df = pandas.read_table(url)\n",
    "doaf_df = doaf_df.rename(columns={'doid_code': 'doid_id', 'GeneID': 'entrez_gene_id'})\n",
    "doaf_df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>doid_id</th>\n",
       "      <th>doid_name</th>\n",
       "      <th>entrez_gene_id</th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>count</th>\n",
       "      <th>pubmeds_max</th>\n",
       "      <th>score_max</th>\n",
       "      <th>score_mean</th>\n",
       "      <th>associationType</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>idiopathic pulmonary fibrosis</td>\n",
       "      <td>729238</td>\n",
       "      <td>SFTPA2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.620284</td>\n",
       "      <td>0.620284</td>\n",
       "      <td>Biomarker|GeneticVariation</td>\n",
       "      <td>BeFree|CLINVAR|CTD_human|UNIPROT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>idiopathic pulmonary fibrosis</td>\n",
       "      <td>7015</td>\n",
       "      <td>TERT</td>\n",
       "      <td>1.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.422153</td>\n",
       "      <td>0.422153</td>\n",
       "      <td>Biomarker|GeneticVariation</td>\n",
       "      <td>BeFree|CLINVAR|CTD_human|GAD|LHGDN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        doid_id                      doid_name  entrez_gene_id geneSymbol  \\\n",
       "0  DOID:0050156  idiopathic pulmonary fibrosis          729238     SFTPA2   \n",
       "1  DOID:0050156  idiopathic pulmonary fibrosis            7015       TERT   \n",
       "\n",
       "   count  pubmeds_max  score_max  score_mean             associationType  \\\n",
       "0    1.0          1.0   0.620284    0.620284  Biomarker|GeneticVariation   \n",
       "1    1.0         10.0   0.422153    0.422153  Biomarker|GeneticVariation   \n",
       "\n",
       "                               source  \n",
       "0    BeFree|CLINVAR|CTD_human|UNIPROT  \n",
       "1  BeFree|CLINVAR|CTD_human|GAD|LHGDN  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# DisGeNET\n",
    "commit = 'fdc5f42f2da745cbf71d7b4cc5021de5685e4a11'\n",
    "#url = utils.rawgit('dhimmel', 'disgenet', commit, 'data/consolidated.tsv')\n",
    "url = rawgit + 'dhimmel/' + 'disgenet/' + commit + '/data/consolidated.tsv'\n",
    "disgenet_df = pandas.read_table(url)\n",
    "disgenet_df = disgenet_df.rename(columns={'doid_code': 'doid_id', 'geneId': 'entrez_gene_id'})\n",
    "disgenet_df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>doid_id</th>\n",
       "      <th>doid_name</th>\n",
       "      <th>locus</th>\n",
       "      <th>high_confidence</th>\n",
       "      <th>primary</th>\n",
       "      <th>status</th>\n",
       "      <th>entrez_gene_id</th>\n",
       "      <th>symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DOID:9970</td>\n",
       "      <td>obesity</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>HC-P</td>\n",
       "      <td>3953</td>\n",
       "      <td>LEPR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DOID:9970</td>\n",
       "      <td>obesity</td>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>HC-P</td>\n",
       "      <td>4094</td>\n",
       "      <td>MAF</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     doid_id doid_name  locus  high_confidence  primary status  \\\n",
       "0  DOID:9970   obesity      0                1        1   HC-P   \n",
       "1  DOID:9970   obesity     14                1        1   HC-P   \n",
       "\n",
       "   entrez_gene_id symbol  \n",
       "0            3953   LEPR  \n",
       "1            4094    MAF  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# hetio GWAS\n",
    "commit = '0617ea7ea8268f21f5ca1b8dbe487dd12671fc7b'\n",
    "#url = utils.rawgit('dhimmel', 'gwas-catalog', commit, 'data/gene-associations.tsv')\n",
    "url = rawgit + 'dhimmel/' + 'gwas-catalog/' + commit + '/data/gene-associations.tsv'\n",
    "gwas_df = pandas.read_table(url)\n",
    "gwas_df = gwas_df.rename(columns={'doid_code': 'doid_id', 'gene': 'entrez_gene_id'})\n",
    "gwas_df.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Filters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "orpha_df = orpha_df.query('gene_type == \"gene with protein product\" or dga_status == \"Assessed\"')\n",
    "diseases_df = diseases_df.query('score_integrated_no_distild >= 2')\n",
    "doaf_df = doaf_df.query('count >= 3')\n",
    "disgenet_df = disgenet_df.query('score_max >= 0.06')\n",
    "gwas_df = gwas_df[gwas_df.status == 'HC-P']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Combine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "orpha_df['provenance'] = 'Orphanet'\n",
    "diseases_df['provenance'] = 'DISEASES'\n",
    "doaf_df['provenance'] = 'DOAF'\n",
    "disgenet_df['provenance'] = 'DisGeNET'\n",
    "gwas_df['provenance'] = 'GWAS Catalog'\n",
    "\n",
    "orpha_df['license'] = 'CC BY-ND 3.0'\n",
    "diseases_df['license'] = 'CC BY 4.0'\n",
    "doaf_df['license'] = ''\n",
    "disgenet_df['license'] = 'ODbL 1.0'\n",
    "gwas_df['license'] = 'CC BY 4.0'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "DisGeNET        7552\n",
       "DISEASES        4990\n",
       "DOAF            1649\n",
       "GWAS Catalog    1283\n",
       "Name: provenance, dtype: int64"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs = [df[['doid_id', 'entrez_gene_id', 'provenance', 'license']]\n",
    "       for df in (diseases_df, doaf_df, disgenet_df, gwas_df)]\n",
    "concat_df = pandas.concat(dfs)\n",
    "concat_df = disease_df.merge(gene_df.merge(concat_df))\n",
    "concat_df.provenance.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "DisGeNET        7552\n",
       "Orphanet        6609\n",
       "DISEASES        4990\n",
       "DOAF            1649\n",
       "GWAS Catalog    1283\n",
       "Name: provenance, dtype: int64"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concat_df.head(2)\n",
    "orpha_df = orpha_df[['doid_id', 'doid_name', 'entrez_gene_id', 'gene_symbol', 'provenance', 'license']]\n",
    "concat_df = pandas.concat([concat_df,orpha_df])\n",
    "concat_df.provenance.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>doid_id</th>\n",
       "      <th>doid_name</th>\n",
       "      <th>entrez_gene_id</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>sources</th>\n",
       "      <th>license</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DOID:2531</td>\n",
       "      <td>Hematologic cancer</td>\n",
       "      <td>25</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>DISEASES|DisGeNET</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DOID:2531</td>\n",
       "      <td>Hematologic cancer</td>\n",
       "      <td>27</td>\n",
       "      <td>ABL2</td>\n",
       "      <td>DisGeNET</td>\n",
       "      <td>ODbL 1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DOID:2531</td>\n",
       "      <td>Hematologic cancer</td>\n",
       "      <td>54</td>\n",
       "      <td>ACP5</td>\n",
       "      <td>DISEASES</td>\n",
       "      <td>CC BY 4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DOID:2531</td>\n",
       "      <td>Hematologic cancer</td>\n",
       "      <td>113</td>\n",
       "      <td>ADCY7</td>\n",
       "      <td>DisGeNET</td>\n",
       "      <td>ODbL 1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>DOID:2531</td>\n",
       "      <td>Hematologic cancer</td>\n",
       "      <td>142</td>\n",
       "      <td>PARP1</td>\n",
       "      <td>DISEASES|DisGeNET</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     doid_id           doid_name  entrez_gene_id gene_symbol  \\\n",
       "0  DOID:2531  Hematologic cancer              25        ABL1   \n",
       "1  DOID:2531  Hematologic cancer              27        ABL2   \n",
       "2  DOID:2531  Hematologic cancer              54        ACP5   \n",
       "3  DOID:2531  Hematologic cancer             113       ADCY7   \n",
       "4  DOID:2531  Hematologic cancer             142       PARP1   \n",
       "\n",
       "             sources    license  \n",
       "0  DISEASES|DisGeNET       None  \n",
       "1           DisGeNET   ODbL 1.0  \n",
       "2           DISEASES  CC BY 4.0  \n",
       "3           DisGeNET   ODbL 1.0  \n",
       "4  DISEASES|DisGeNET       None  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def condense(df):\n",
    "    \"\"\"Consolidate multiple associations into a single Series.\"\"\"\n",
    "    row = pandas.Series()\n",
    "    row['sources'] = '|'.join(df.provenance)\n",
    "    licenses = set(df.license)\n",
    "    licenses.discard('')\n",
    "    try:\n",
    "        row['license'], = licenses\n",
    "    except ValueError:\n",
    "        row['license'] = None\n",
    "    return row\n",
    "\n",
    "short_df = concat_df.groupby(['doid_id', 'entrez_gene_id']).apply(condense).reset_index()\n",
    "short_df = disease_df.merge(gene_df.merge(short_df))\n",
    "short_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "short_df.to_csv('DaG-association.tsv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "short_df = concat_df.groupby(['doid_id', 'entrez_gene_id']).apply(condense).reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>doid_id</th>\n",
       "      <th>entrez_gene_id</th>\n",
       "      <th>sources</th>\n",
       "      <th>license</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>1832.0</td>\n",
       "      <td>DisGeNET</td>\n",
       "      <td>ODbL 1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>2335.0</td>\n",
       "      <td>DISEASES</td>\n",
       "      <td>CC BY 4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>2355.0</td>\n",
       "      <td>DisGeNET</td>\n",
       "      <td>ODbL 1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>4088.0</td>\n",
       "      <td>DISEASES</td>\n",
       "      <td>CC BY 4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>5328.0</td>\n",
       "      <td>DisGeNET</td>\n",
       "      <td>ODbL 1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>6440.0</td>\n",
       "      <td>DISEASES|DisGeNET</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>6441.0</td>\n",
       "      <td>DISEASES</td>\n",
       "      <td>CC BY 4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>7015.0</td>\n",
       "      <td>DISEASES|DOAF|DisGeNET|GWAS Catalog</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>7040.0</td>\n",
       "      <td>DISEASES</td>\n",
       "      <td>CC BY 4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>10144.0</td>\n",
       "      <td>DisGeNET</td>\n",
       "      <td>ODbL 1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>23250.0</td>\n",
       "      <td>DisGeNET</td>\n",
       "      <td>ODbL 1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>54472.0</td>\n",
       "      <td>GWAS Catalog</td>\n",
       "      <td>CC BY 4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>79991.0</td>\n",
       "      <td>DisGeNET</td>\n",
       "      <td>ODbL 1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>91039.0</td>\n",
       "      <td>DisGeNET</td>\n",
       "      <td>ODbL 1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>255520.0</td>\n",
       "      <td>DISEASES|DisGeNET</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>653509.0</td>\n",
       "      <td>DisGeNET</td>\n",
       "      <td>ODbL 1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>727897.0</td>\n",
       "      <td>DisGeNET</td>\n",
       "      <td>ODbL 1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>DOID:0050156</td>\n",
       "      <td>729238.0</td>\n",
       "      <td>DISEASES|DisGeNET</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>DOID:0050425</td>\n",
       "      <td>1775.0</td>\n",
       "      <td>DISEASES</td>\n",
       "      <td>CC BY 4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>DOID:0050425</td>\n",
       "      <td>4211.0</td>\n",
       "      <td>DISEASES|DisGeNET|GWAS Catalog</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>DOID:0050425</td>\n",
       "      <td>5348.0</td>\n",
       "      <td>DISEASES</td>\n",
       "      <td>CC BY 4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>DOID:0050425</td>\n",
       "      <td>5443.0</td>\n",
       "      <td>DisGeNET</td>\n",
       "      <td>ODbL 1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>DOID:0050425</td>\n",
       "      <td>5607.0</td>\n",
       "      <td>DISEASES|GWAS Catalog</td>\n",
       "      <td>CC BY 4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>DOID:0050425</td>\n",
       "      <td>5789.0</td>\n",
       "      <td>DISEASES|DisGeNET|GWAS Catalog</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>DOID:0050425</td>\n",
       "      <td>7018.0</td>\n",
       "      <td>DisGeNET</td>\n",
       "      <td>ODbL 1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>DOID:0050425</td>\n",
       "      <td>27324.0</td>\n",
       "      <td>GWAS Catalog</td>\n",
       "      <td>CC BY 4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>DOID:0050425</td>\n",
       "      <td>51072.0</td>\n",
       "      <td>DISEASES</td>\n",
       "      <td>CC BY 4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>DOID:0050425</td>\n",
       "      <td>94033.0</td>\n",
       "      <td>DISEASES</td>\n",
       "      <td>CC BY 4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>DOID:0050425</td>\n",
       "      <td>114781.0</td>\n",
       "      <td>DISEASES|DisGeNET|GWAS Catalog</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>DOID:0050425</td>\n",
       "      <td>348654.0</td>\n",
       "      <td>DISEASES</td>\n",
       "      <td>CC BY 4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19175</th>\n",
       "      <td>ORPHA:99943</td>\n",
       "      <td>4359.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19176</th>\n",
       "      <td>ORPHA:99944</td>\n",
       "      <td>54332.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19177</th>\n",
       "      <td>ORPHA:99945</td>\n",
       "      <td>26353.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19178</th>\n",
       "      <td>ORPHA:99946</td>\n",
       "      <td>23095.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19179</th>\n",
       "      <td>ORPHA:99947</td>\n",
       "      <td>9927.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19180</th>\n",
       "      <td>ORPHA:99948</td>\n",
       "      <td>54332.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19181</th>\n",
       "      <td>ORPHA:99949</td>\n",
       "      <td>79628.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19182</th>\n",
       "      <td>ORPHA:99950</td>\n",
       "      <td>10397.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19183</th>\n",
       "      <td>ORPHA:99951</td>\n",
       "      <td>1959.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19184</th>\n",
       "      <td>ORPHA:99952</td>\n",
       "      <td>57716.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19185</th>\n",
       "      <td>ORPHA:99953</td>\n",
       "      <td>3098.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19186</th>\n",
       "      <td>ORPHA:99954</td>\n",
       "      <td>121512.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19187</th>\n",
       "      <td>ORPHA:99955</td>\n",
       "      <td>8898.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19188</th>\n",
       "      <td>ORPHA:99956</td>\n",
       "      <td>81846.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19189</th>\n",
       "      <td>ORPHA:99960</td>\n",
       "      <td>5205.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19190</th>\n",
       "      <td>ORPHA:99961</td>\n",
       "      <td>8647.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19191</th>\n",
       "      <td>ORPHA:99966</td>\n",
       "      <td>6598.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19192</th>\n",
       "      <td>ORPHA:99967</td>\n",
       "      <td>1649.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19193</th>\n",
       "      <td>ORPHA:99967</td>\n",
       "      <td>2521.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19194</th>\n",
       "      <td>ORPHA:99970</td>\n",
       "      <td>1019.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19195</th>\n",
       "      <td>ORPHA:99970</td>\n",
       "      <td>4193.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19196</th>\n",
       "      <td>ORPHA:99970</td>\n",
       "      <td>8091.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19197</th>\n",
       "      <td>ORPHA:99971</td>\n",
       "      <td>1019.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19198</th>\n",
       "      <td>ORPHA:99971</td>\n",
       "      <td>4193.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19199</th>\n",
       "      <td>ORPHA:99971</td>\n",
       "      <td>8091.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19200</th>\n",
       "      <td>ORPHA:99977</td>\n",
       "      <td>6049.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19201</th>\n",
       "      <td>ORPHA:99977</td>\n",
       "      <td>7048.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19202</th>\n",
       "      <td>ORPHA:99977</td>\n",
       "      <td>9940.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19203</th>\n",
       "      <td>ORPHA:99977</td>\n",
       "      <td>51741.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19204</th>\n",
       "      <td>ORPHA:99989</td>\n",
       "      <td>3767.0</td>\n",
       "      <td>Orphanet</td>\n",
       "      <td>CC BY-ND 3.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19205 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            doid_id  entrez_gene_id                              sources  \\\n",
       "0      DOID:0050156          1832.0                             DisGeNET   \n",
       "1      DOID:0050156          2335.0                             DISEASES   \n",
       "2      DOID:0050156          2355.0                             DisGeNET   \n",
       "3      DOID:0050156          4088.0                             DISEASES   \n",
       "4      DOID:0050156          5328.0                             DisGeNET   \n",
       "5      DOID:0050156          6440.0                    DISEASES|DisGeNET   \n",
       "6      DOID:0050156          6441.0                             DISEASES   \n",
       "7      DOID:0050156          7015.0  DISEASES|DOAF|DisGeNET|GWAS Catalog   \n",
       "8      DOID:0050156          7040.0                             DISEASES   \n",
       "9      DOID:0050156         10144.0                             DisGeNET   \n",
       "10     DOID:0050156         23250.0                             DisGeNET   \n",
       "11     DOID:0050156         54472.0                         GWAS Catalog   \n",
       "12     DOID:0050156         79991.0                             DisGeNET   \n",
       "13     DOID:0050156         91039.0                             DisGeNET   \n",
       "14     DOID:0050156        255520.0                    DISEASES|DisGeNET   \n",
       "15     DOID:0050156        653509.0                             DisGeNET   \n",
       "16     DOID:0050156        727897.0                             DisGeNET   \n",
       "17     DOID:0050156        729238.0                    DISEASES|DisGeNET   \n",
       "18     DOID:0050425          1775.0                             DISEASES   \n",
       "19     DOID:0050425          4211.0       DISEASES|DisGeNET|GWAS Catalog   \n",
       "20     DOID:0050425          5348.0                             DISEASES   \n",
       "21     DOID:0050425          5443.0                             DisGeNET   \n",
       "22     DOID:0050425          5607.0                DISEASES|GWAS Catalog   \n",
       "23     DOID:0050425          5789.0       DISEASES|DisGeNET|GWAS Catalog   \n",
       "24     DOID:0050425          7018.0                             DisGeNET   \n",
       "25     DOID:0050425         27324.0                         GWAS Catalog   \n",
       "26     DOID:0050425         51072.0                             DISEASES   \n",
       "27     DOID:0050425         94033.0                             DISEASES   \n",
       "28     DOID:0050425        114781.0       DISEASES|DisGeNET|GWAS Catalog   \n",
       "29     DOID:0050425        348654.0                             DISEASES   \n",
       "...             ...             ...                                  ...   \n",
       "19175   ORPHA:99943          4359.0                             Orphanet   \n",
       "19176   ORPHA:99944         54332.0                             Orphanet   \n",
       "19177   ORPHA:99945         26353.0                             Orphanet   \n",
       "19178   ORPHA:99946         23095.0                             Orphanet   \n",
       "19179   ORPHA:99947          9927.0                             Orphanet   \n",
       "19180   ORPHA:99948         54332.0                             Orphanet   \n",
       "19181   ORPHA:99949         79628.0                             Orphanet   \n",
       "19182   ORPHA:99950         10397.0                             Orphanet   \n",
       "19183   ORPHA:99951          1959.0                             Orphanet   \n",
       "19184   ORPHA:99952         57716.0                             Orphanet   \n",
       "19185   ORPHA:99953          3098.0                             Orphanet   \n",
       "19186   ORPHA:99954        121512.0                             Orphanet   \n",
       "19187   ORPHA:99955          8898.0                             Orphanet   \n",
       "19188   ORPHA:99956         81846.0                             Orphanet   \n",
       "19189   ORPHA:99960          5205.0                             Orphanet   \n",
       "19190   ORPHA:99961          8647.0                             Orphanet   \n",
       "19191   ORPHA:99966          6598.0                             Orphanet   \n",
       "19192   ORPHA:99967          1649.0                             Orphanet   \n",
       "19193   ORPHA:99967          2521.0                             Orphanet   \n",
       "19194   ORPHA:99970          1019.0                             Orphanet   \n",
       "19195   ORPHA:99970          4193.0                             Orphanet   \n",
       "19196   ORPHA:99970          8091.0                             Orphanet   \n",
       "19197   ORPHA:99971          1019.0                             Orphanet   \n",
       "19198   ORPHA:99971          4193.0                             Orphanet   \n",
       "19199   ORPHA:99971          8091.0                             Orphanet   \n",
       "19200   ORPHA:99977          6049.0                             Orphanet   \n",
       "19201   ORPHA:99977          7048.0                             Orphanet   \n",
       "19202   ORPHA:99977          9940.0                             Orphanet   \n",
       "19203   ORPHA:99977         51741.0                             Orphanet   \n",
       "19204   ORPHA:99989          3767.0                             Orphanet   \n",
       "\n",
       "            license  \n",
       "0          ODbL 1.0  \n",
       "1         CC BY 4.0  \n",
       "2          ODbL 1.0  \n",
       "3         CC BY 4.0  \n",
       "4          ODbL 1.0  \n",
       "5              None  \n",
       "6         CC BY 4.0  \n",
       "7              None  \n",
       "8         CC BY 4.0  \n",
       "9          ODbL 1.0  \n",
       "10         ODbL 1.0  \n",
       "11        CC BY 4.0  \n",
       "12         ODbL 1.0  \n",
       "13         ODbL 1.0  \n",
       "14             None  \n",
       "15         ODbL 1.0  \n",
       "16         ODbL 1.0  \n",
       "17             None  \n",
       "18        CC BY 4.0  \n",
       "19             None  \n",
       "20        CC BY 4.0  \n",
       "21         ODbL 1.0  \n",
       "22        CC BY 4.0  \n",
       "23             None  \n",
       "24         ODbL 1.0  \n",
       "25        CC BY 4.0  \n",
       "26        CC BY 4.0  \n",
       "27        CC BY 4.0  \n",
       "28             None  \n",
       "29        CC BY 4.0  \n",
       "...             ...  \n",
       "19175  CC BY-ND 3.0  \n",
       "19176  CC BY-ND 3.0  \n",
       "19177  CC BY-ND 3.0  \n",
       "19178  CC BY-ND 3.0  \n",
       "19179  CC BY-ND 3.0  \n",
       "19180  CC BY-ND 3.0  \n",
       "19181  CC BY-ND 3.0  \n",
       "19182  CC BY-ND 3.0  \n",
       "19183  CC BY-ND 3.0  \n",
       "19184  CC BY-ND 3.0  \n",
       "19185  CC BY-ND 3.0  \n",
       "19186  CC BY-ND 3.0  \n",
       "19187  CC BY-ND 3.0  \n",
       "19188  CC BY-ND 3.0  \n",
       "19189  CC BY-ND 3.0  \n",
       "19190  CC BY-ND 3.0  \n",
       "19191  CC BY-ND 3.0  \n",
       "19192  CC BY-ND 3.0  \n",
       "19193  CC BY-ND 3.0  \n",
       "19194  CC BY-ND 3.0  \n",
       "19195  CC BY-ND 3.0  \n",
       "19196  CC BY-ND 3.0  \n",
       "19197  CC BY-ND 3.0  \n",
       "19198  CC BY-ND 3.0  \n",
       "19199  CC BY-ND 3.0  \n",
       "19200  CC BY-ND 3.0  \n",
       "19201  CC BY-ND 3.0  \n",
       "19202  CC BY-ND 3.0  \n",
       "19203  CC BY-ND 3.0  \n",
       "19204  CC BY-ND 3.0  \n",
       "\n",
       "[19205 rows x 4 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "short_df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
